### =========================================================================
### excluderanges metadata
### -------------------------------------------------------------------------
###

# Empty data frame
# meta <- data.frame(
#     Title = c(),
#     Description = c(),
#     BiocVersion = rep(),
#     Genome = c(),
#     SourceType = rep(),
#     SourceUrl = rep(),
#     SourceVersion = rep(),
#     Species = c(),
#     TaxonomyId = c(),
#     Coordinate_1_based = c(),
#     DataProvider = c(),
#     Maintainer = rep(),
#     RDataClass = rep(),
#     DispatchClass = rep(),
#     RDataPath = c(),
#     Tags = c(),
#     Notes = "")

# # Original (42) and NEW (82) objects (124 total)
# meta <- data.frame(
#   Title = c("ce10.Kundaje.ce10-Excludable.rds", 
#             "dm3.Kundaje.dm3-Excludable.rds", 
#             "hg19.Bernstein.Mint_Excludable_hg19.rds", 
#             "hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds", 
#             "hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds", 
#             "hg19.Wold.hg19mitoExcludable.rds", 
#             "hg19.Yeo.eCLIP_Excludableregions.hg19.rds", 
#             "hg38.Bernstein.Mint_Excludable_GRCh38.rds", 
#             "hg38.Kundaje.GRCh38.Excludable.rds", 
#             "hg38.Kundaje.GRCh38_unified_Excludable.rds", 
#             "hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds", 
#             "hg38.Wold.hg38mitoExcludable.rds", 
#             "hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds", 
#             "mm10.Hardison.Excludable.full.rds", 
#             "mm10.Hardison.psuExcludable.mm10.rds", 
#             "mm10.Kundaje.anshul.Excludable.mm10.rds", 
#             "mm10.Kundaje.mm10.Excludable.rds", 
#             "mm10.Wold.mm10mitoExcludable.rds", 
#             "mm9.Wold.mm9mitoExcludable.rds",
#             "hg19.UCSC.centromere.rds", 
#             "hg19.UCSC.clone.rds", 
#             "hg19.UCSC.contig.rds", 
#             "hg19.UCSC.heterochromatin.rds", 
#             "hg19.UCSC.scaffold.rds", 
#             "hg19.UCSC.short_arm.rds", 
#             "hg19.UCSC.telomere.rds", 
#             "hg38.UCSC.contig.rds", 
#             "hg38.UCSC.heterochromatin.rds", 
#             "hg38.UCSC.scaffold.rds", 
#             "hg38.UCSC.short_arm.rds", 
#             "hg38.UCSC.telomere.rds", 
#             "mm10.UCSC.centromere.rds", 
#             "mm10.UCSC.clone.rds", 
#             "mm10.UCSC.contig.rds", 
#             "mm10.UCSC.fragment.rds", 
#             "mm10.UCSC.other.rds", 
#             "mm10.UCSC.short_arm.rds", 
#             "mm10.UCSC.telomere.rds", 
#             "mm9.UCSC.centromere.rds", 
#             "mm9.UCSC.contig.rds", 
#             "mm9.UCSC.fragment.rds",
#             "mm10.UCSC.scaffold.rds",
#             "T2T.excluderanges",
#             "hg38.Kundaje.GRCh38_unified_Excludable",
#             "hg38.Bernstein.Mint_Excludable_GRCh38",
#             "hg38.Boyle.hg38-Excludable.v2",
#             "hg38.Kundaje.GRCh38.Excludable",
#             "hg38.Lareau.hg38.full.Excludable",
#             "hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38",
#             "hg38.Wimberley.peakPass60Perc_sorted",
#             "hg38.Wold.hg38mitoExcludable",
#             "hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed",
#             "hg19.Boyle.hg19-Excludable.v2",
#             "hg19.Bernstein.Mint_Excludable_hg19",
#             "hg19.Birney.wgEncodeDacMapabilityConsensusExcludable",
#             "hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable",
#             "hg19.Lareau.hg19.full.Excludable",
#             "hg19.Wold.hg19mitoExcludable",
#             "hg19.Yeo.eCLIP_Excludableregions.hg19",
#             "mm39.excluderanges",
#             "mm10.Boyle.mm10-Excludable.v2",
#             "mm10.Hardison.Excludable.full",
#             "mm10.Hardison.psuExcludable.mm10",
#             "mm10.Kundaje.anshul.Excludable.mm10",
#             "mm10.Kundaje.mm10.Excludable",
#             "mm10.Lareau.mm10.full.Excludable",
#             "mm10.Wold.mm10mitoExcludable",
#             "mm9.Lareau.mm9.full.Excludable",
#             "mm9.Wold.mm9mitoExcludable",
#             "ce11.Boyle.ce11-Excludable.v2",
#             "ce10.Boyle.ce10-Excludable.v2",
#             "ce10.Kundaje.ce10-Excludable",
#             "danRer10.Domingues.Excludableed",
#             "danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome",
#             "dm6.Boyle.dm6-Excludable.v2",
#             "dm3.Boyle.dm3-Excludable.v2",
#             "dm3.Kundaje.dm3-Excludable",
#             "TAIR10.Wimberley.predicted_excluded_list_sorted_0.6",
#             "TAIR10.Klasfeld.arabidopsis_Excludable_20inputs",
#             "TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs",
#             "T2T.Lareau.chm13v2.0_peaks",
#             "hg38.Lareau.hg38_peaks",
#             "hg19.Lareau.hg19_peaks",
#             "mm10.Lareau.mm10_peaks",
#             "mm9.Lareau.mm9_peaks",
#             "hg19.UCSC.numtS",
#             "mm9.UCSC.numtS",
#             "T2T.CHM13.chm13.draft_v2.0.cen_mask",
#             "T2T.CHM13.chm13.draft_v1.1.telomere",
#             "T2T.UCSC.censat",
#             "T2T.UCSC.gap",
#             "T2T.UCSC.hgUnique.hg38",
#             "hg38.UCSC.centromere",
#             "hg38.UCSC.telomere",
#             "hg38.UCSC.short_arm",
#             "hg38.UCSC.heterochromatin",
#             "hg38.UCSC.contig",
#             "hg38.UCSC.scaffold",
#             "hg19.UCSC.centromere",
#             "hg19.UCSC.telomere",
#             "hg19.UCSC.short_arm",
#             "hg19.UCSC.heterochromatin",
#             "hg19.UCSC.clone",
#             "hg19.UCSC.contig",
#             "mm39.UCSC.centromere",
#             "mm39.UCSC.telomere",
#             "mm39.UCSC.short_arm",
#             "mm39.UCSC.contig",
#             "mm39.UCSC.scaffold",
#             "mm10.UCSC.centromere",
#             "mm10.UCSC.telomere",
#             "mm10.UCSC.short_arm",
#             "mm10.UCSC.clone",
#             "mm10.UCSC.contig",
#             "mm10.UCSC.other",
#             "mm9.UCSC.centromere",
#             "mm9.UCSC.fragment",
#             "mm9.UCSC.contig",
#             "danRer10.UCSC.contig",
#             "danRer10.UCSC.scaffold",
#             "dm6.UCSC.other",
#             "dm3.UCSC.contig",
#             "dm3.UCSC.scaffold",
#             "TAIR10.UCSC.araTha1.gap"),
#   Description = c("Exclusion regions for ce10 by Anshul Kundaje, Stanford", 
#                   "Exclusion regions for dm3 by Anshul Kundaje, Stanford", 
#                   "Exclusion regions for hg19 by Bradley Bernstein, Broad", 
#                   "Exclusion regions for hg19 by Ewan Birney, EBI", 
#                   "Exclusion regions for hg19 by Gregory Crawford, Duke", 
#                   "Exclusion regions for hg19 by Barbara Wold, Caltech", 
#                   "Exclusion regions for hg19 by Gene Yeo, UCSD", 
#                   "Exclusion regions for hg38 by Bradley Bernstein, Broad", 
#                   "Exclusion regions for hg38 by Anshul Kundaje, Stanford", 
#                   "Exclusion regions for hg38 by Anshul Kundaje, Stanford", 
#                   "Exclusion regions for hg38 by Tim Reddy, Duke", 
#                   "Exclusion regions for hg38 by Barbara Wold, Caltech", 
#                   "Exclusion regions for hg38 by Gene Yeo, UCSD", 
#                   "Exclusion regions for mm10 by Ross Hardison, PennState", 
#                   "Exclusion regions for mm10 by Ross Hardison, PennState", 
#                   "Exclusion regions for mm10 by Anshul Kundaje, Stanford", 
#                   "Exclusion regions for mm10 by Anshul Kundaje, Stanford", 
#                   "Exclusion regions for mm10 by Barbara Wold, Caltech", 
#                   "Exclusion regions for mm9 by Barbara Wold, Caltech",
#                   "hg19 centromere from the UCSC gap table", 
#                   "hg19 clone from the UCSC gap table", 
#                   "hg19 contig from the UCSC gap table", 
#                   "hg19 heterochromatin from the UCSC gap table", 
#                   "hg19 scaffold from the UCSC gap table", 
#                   "hg19 short_arm from the UCSC gap table", 
#                   "hg19 telomere from the UCSC gap table", 
#                   "hg38 contig from the UCSC gap table", 
#                   "hg38 heterochromatin from the UCSC gap table", 
#                   "hg38 scaffold from the UCSC gap table", 
#                   "hg38 short_arm from the UCSC gap table", 
#                   "hg38 telomere from the UCSC gap table", 
#                   "mm10 centromere from the UCSC gap table", 
#                   "mm10 clone from the UCSC gap table", 
#                   "mm10 contig from the UCSC gap table", 
#                   "mm10 fragment from the UCSC gap table", 
#                   "mm10 other from the UCSC gap table", 
#                   "mm10 short_arm from the UCSC gap table", 
#                   "mm10 telomere from the UCSC gap table", 
#                   "mm9 centromere from the UCSC gap table", 
#                   "mm9 contig from the UCSC gap table", 
#                   "mm9 fragment from the UCSC gap table",
#                   "mm10 scaffold from the UCSC gap table",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined as a combination of hg38.Lareau.hg38_peaks, hg38.Boyle.hg38-Excludable.v2, and hg38.Wimberley.peakPass60Perc_sorted, followed by manual curation. Supersedes hg38.Kundaje.GRCh38.Excludable.",
#                   "Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
#                   "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
#                   "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Defined by the ewimberley/peakPass software",
#                   "Definition method unknown",
#                   "Defined from eCLIP data",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data",
#                   "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands",
#                   "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), human satellite repeat HSATII, chrM, ribosomal subunit consensus sequences LSU-rRNA_Hsa, SSU-rRNA_Hsa. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
#                   "Definition method unknown",
#                   "Defined from eCLIP data, includes skyscraper, rRNA pseudogene, unreliably mapped satellite repeat, and low complexity skyscraper peak regions",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Definition method unknown",
#                   "Definition method unknown",
#                   "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
#                   "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
#                   "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
#                   "Definition method unknown",
#                   "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
#                   "Definition method unknown",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined by Anshul Kundaje, superseded by ce10.Boyle.ce10-Excludable.v2",
#                   "Defined manually using total RNA-seq.",
#                   "Defined via MACS2 peak calling using ChIP-seq (PMID: 33239788)",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
#                   "Defined by Anshul Kundaje. Contains heterochromatin chromosomes chr2LHet. Superseded by dm3.Boyle.dm3-Excludable.v2",
#                   "Defined by the ewimberley/peakPass software",
#                   "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions (DOI: 10.1101/2022.02.27.482177)",
#                   "Defined by the green screen pipeline (DOI: 10.1101/2022.02.27.482177)",
#                   "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
#                   "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
#                   "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
#                   "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
#                   "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
#                   "Human NumtS mitochondrial sequence",
#                   "Mouse NumtS mitochondrial sequence",
#                   "Centromeric satellite masking bed file (v2.0)",
#                   "Telomere identified by the VGP pipeline (v1.1)",
#                   "T2T peri/centromeric satellite annotation (v2.0, 20220329, CHM13 v2.0)",
#                   "Locations of assembly gaps, as determine by strings of 'N' characters (v1.0)",
#                   "Regions unique to the T2T-CHM13 v2.0 assembly compared to the GRCh38/hg38 and GRCh37/hg19 reference assemblies",
#                   "Gaps from centromeres",
#                   "Gaps from telomeres",
#                   "Gaps on the short arm of the chromosome",
#                   "Gaps from large blocks of heterochromatin",
#                   "Gaps between contigs in scaffolds",
#                   "Gaps between scaffolds in chromosome assemblies. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Gaps from centromeres",
#                   "Gaps from telomeres",
#                   "Gaps on the short arm of the chromosome",
#                   "Gaps from large blocks of heterochromatin",
#                   "Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Gaps between contigs in scaffolds",
#                   "Gaps from centromeres",
#                   "Gaps from telomeres",
#                   "Gaps on the short arm of the chromosome",
#                   "Gaps between contigs in scaffolds",
#                   "Gaps between scaffolds in chromosome assemblies",
#                   "Gaps from centromeres",
#                   "Gaps from telomeres",
#                   "Gaps on the short arm of the chromosome",
#                   "Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Gaps between contigs in scaffolds",
#                   "Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Gaps from centromeres",
#                   "Gaps between the contigs of a draft clone. (In this context, a contig is a set of overlapping sequence reads). Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Gaps between contigs in scaffolds. Has extra chromosomes, use keepStandardChromosomes() filtering",
#                   "Gaps between contigs in scaffolds",
#                   "Gaps between scaffolds in chromosome assemblies",
#                   "Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file",
#                   "Gaps between contigs in scaffolds",
#                   "Gaps between scaffolds in chromosome assemblies",
#                   "Gaps in the May 2011 Arabidopsis thaliana genome assembly"),
#   BiocVersion = c(rep("3.14", 42), rep("3.16", 82)),
#   Genome = c("ce10", 
#              "dm3", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm9",
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg19", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "hg38", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm10", 
#              "mm9", 
#              "mm9", 
#              "mm9",
#              "mm10",
#              "T2T-CHM13",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "mm39",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm9",
#              "mm9",
#              "ce11",
#              "ce10",
#              "ce10",
#              "danRer10",
#              "danRer10",
#              "dm6",
#              "dm3",
#              "dm3",
#              "TAIR10",
#              "TAIR10",
#              "TAIR10",
#              "T2T-CHM13",
#              "hg38",
#              "hg19",
#              "mm10",
#              "mm9",
#              "hg19",
#              "mm9",
#              "T2T-CHM13",
#              "T2T-CHM13",
#              "T2T-CHM13",
#              "T2T-CHM13",
#              "T2T-CHM13",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg38",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "hg19",
#              "mm39",
#              "mm39",
#              "mm39",
#              "mm39",
#              "mm39",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm10",
#              "mm9",
#              "mm9",
#              "mm9",
#              "danRer10",
#              "danRer10",
#              "dm6",
#              "dm3",
#              "dm3",
#              "TAIR10"),
#   SourceType = c(rep("RDS", 42), rep("RDS", 82)),
#   SourceUrl = c(rep("https://drive.google.com/drive/folders/1MFn6LPZD1zZRQz7biR0Fyl-mzkx4CIxS?usp=sharing", 42),
#                 rep("https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2kfHOWFBSJT?usp=sharing", 82)),
#   SourceVersion = c(rep("August 22 2021", 42), rep("September 22 2022", 82)),
#   Species = c(c("Caenorhabditis elegans", "Drosophila melanogaster", rep("Homo sapiens", 11), rep("Mus musculus", 6), rep("Homo sapiens", 12), rep("Mus musculus", 11)),
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Caenorhabditis elegans",
#               "Caenorhabditis elegans",
#               "Caenorhabditis elegans",
#               "Danio rerio",
#               "Danio rerio",
#               "Drosophila melanogaster",
#               "Drosophila melanogaster",
#               "Drosophila melanogaster",
#               "Arabidopsis thaliana",
#               "Arabidopsis thaliana",
#               "Arabidopsis thaliana",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Mus musculus",
#               "Mus musculus",
#               "Homo sapiens",
#               "Mus musculus",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Homo sapiens",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Mus musculus",
#               "Danio rerio",
#               "Danio rerio",
#               "Drosophila melanogaster",
#               "Drosophila melanogaster",
#               "Drosophila melanogaster",
#               "Arabidopsis thaliana"),
#   TaxonomyId = c(c(6239, 7227, rep(9606, 11), rep(10090, 6), rep(9606, 12), rep(10090, 11)),
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "6239",
#                  "6239",
#                  "6239",
#                  "7955",
#                  "7955",
#                  "7227",
#                  "7227",
#                  "7227",
#                  "3702",
#                  "3702",
#                  "3702",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "10090",
#                  "10090",
#                  "9606",
#                  "10090",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "9606",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "10090",
#                  "7955",
#                  "7955",
#                  "7227",
#                  "7227",
#                  "7227",
#                  "3702"),
#   Coordinate_1_based = c(rep(TRUE, 42),
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          FALSE,
#                          FALSE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          TRUE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          FALSE,
#                          TRUE),
#   DataProvider = c(c(rep("mitra.stanford.edu/kundaje/akundaje/release/blacklists/", 2), rep("ENCODE", 17), rep("UCSC", 23)),
#                    "excluderanges",
#                    "ENCODE",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "ENCODE",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "ENCODE",
#                    "excluderanges",
#                    "GitHub",
#                    "ENCODE",
#                    "ENCODE",
#                    "ENCODE",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "GitHub",
#                    "ENCODE",
#                    "GitHub",
#                    "GitHub",
#                    "Stanford.edu",
#                    "GitHub",
#                    "Publication",
#                    "GitHub",
#                    "GitHub",
#                    "Stanford.edu",
#                    "GitHub",
#                    "GitHub",
#                    "GitHub",
#                    "GitHub",
#                    "GitHub",
#                    "GitHub",
#                    "GitHub",
#                    "GitHub",
#                    "UCSC",
#                    "UCSC",
#                    "CHM13",
#                    "CHM13",
#                    "UCSChub",
#                    "UCSChub",
#                    "UCSChub",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSC",
#                    "UCSChub"),
#   Maintainer = c(rep("Mikhail Dozmorov <mikhail.dozmorov@gmail.com>", 42), rep("Mikhail Dozmorov <mikhail.dozmorov@gmail.com>", 82)),
#   RDataClass = c(rep("GRanges", 42), rep("GRanges", 82)),
#   DispatchClass = c(rep("RDS", 42), rep("RDS", 82)),
#   RDataPath = c("excluderanges/ce10.Kundaje.ce10-Excludable.rds",
#                 "excluderanges/dm3.Kundaje.dm3-Excludable.rds",
#                 "excluderanges/hg19.Bernstein.Mint_Excludable_hg19.rds",
#                 "excluderanges/hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds",
#                 "excluderanges/hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds",
#                 "excluderanges/hg19.Wold.hg19mitoExcludable.rds",
#                 "excluderanges/hg19.Yeo.eCLIP_Excludableregions.hg19.rds",
#                 "excluderanges/hg38.Bernstein.Mint_Excludable_GRCh38.rds",
#                 "excluderanges/hg38.Kundaje.GRCh38.Excludable.rds",
#                 "excluderanges/hg38.Kundaje.GRCh38_unified_Excludable.rds",
#                 "excluderanges/hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds",
#                 "excluderanges/hg38.Wold.hg38mitoExcludable.rds",
#                 "excluderanges/hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds",
#                 "excluderanges/mm10.Hardison.Excludable.full.rds",
#                 "excluderanges/mm10.Hardison.psuExcludable.mm10.rds",
#                 "excluderanges/mm10.Kundaje.anshul.Excludable.mm10.rds",
#                 "excluderanges/mm10.Kundaje.mm10.Excludable.rds",
#                 "excluderanges/mm10.Wold.mm10mitoExcludable.rds",
#                 "excluderanges/mm9.Wold.mm9mitoExcludable.rds",
#                 "excluderanges/hg19.UCSC.centromere.rds",
#                 "excluderanges/hg19.UCSC.clone.rds",
#                 "excluderanges/hg19.UCSC.contig.rds",
#                 "excluderanges/hg19.UCSC.heterochromatin.rds",
#                 "excluderanges/hg19.UCSC.scaffold.rds",
#                 "excluderanges/hg19.UCSC.short_arm.rds",
#                 "excluderanges/hg19.UCSC.telomere.rds",
#                 "excluderanges/hg38.UCSC.contig.rds",
#                 "excluderanges/hg38.UCSC.heterochromatin.rds",
#                 "excluderanges/hg38.UCSC.scaffold.rds",
#                 "excluderanges/hg38.UCSC.short_arm.rds",
#                 "excluderanges/hg38.UCSC.telomere.rds",
#                 "excluderanges/mm10.UCSC.centromere.rds",
#                 "excluderanges/mm10.UCSC.clone.rds",
#                 "excluderanges/mm10.UCSC.contig.rds",
#                 "excluderanges/mm10.UCSC.fragment.rds",
#                 "excluderanges/mm10.UCSC.other.rds",
#                 "excluderanges/mm10.UCSC.short_arm.rds",
#                 "excluderanges/mm10.UCSC.telomere.rds",
#                 "excluderanges/mm9.UCSC.centromere.rds",
#                 "excluderanges/mm9.UCSC.contig.rds",
#                 "excluderanges/mm9.UCSC.fragment.rds",
#                 "excluderanges/mm10.UCSC.scaffold.rds",
#                 "excluderanges/T2T.excluderanges.rds",
#                 "excluderanges/hg38.Kundaje.GRCh38_unified_Excludable.rds",
#                 "excluderanges/hg38.Bernstein.Mint_Excludable_GRCh38.rds",
#                 "excluderanges/hg38.Boyle.hg38-Excludable.v2.rds",
#                 "excluderanges/hg38.Kundaje.GRCh38.Excludable.rds",
#                 "excluderanges/hg38.Lareau.hg38.full.Excludable.rds",
#                 "excluderanges/hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds",
#                 "excluderanges/hg38.Wimberley.peakPass60Perc_sorted.rds",
#                 "excluderanges/hg38.Wold.hg38mitoExcludable.rds",
#                 "excluderanges/hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds",
#                 "excluderanges/hg19.Boyle.hg19-Excludable.v2.rds",
#                 "excluderanges/hg19.Bernstein.Mint_Excludable_hg19.rds",
#                 "excluderanges/hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds",
#                 "excluderanges/hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds",
#                 "excluderanges/hg19.Lareau.hg19.full.Excludable.rds",
#                 "excluderanges/hg19.Wold.hg19mitoExcludable.rds",
#                 "excluderanges/hg19.Yeo.eCLIP_Excludableregions.hg19.rds",
#                 "excluderanges/mm39.excluderanges.rds",
#                 "excluderanges/mm10.Boyle.mm10-Excludable.v2.rds",
#                 "excluderanges/mm10.Hardison.Excludable.full.rds",
#                 "excluderanges/mm10.Hardison.psuExcludable.mm10.rds",
#                 "excluderanges/mm10.Kundaje.anshul.Excludable.mm10.rds",
#                 "excluderanges/mm10.Kundaje.mm10.Excludable.rds",
#                 "excluderanges/mm10.Lareau.mm10.full.Excludable.rds",
#                 "excluderanges/mm10.Wold.mm10mitoExcludable.rds",
#                 "excluderanges/mm9.Lareau.mm9.full.Excludable.rds",
#                 "excluderanges/mm9.Wold.mm9mitoExcludable.rds",
#                 "excluderanges/ce11.Boyle.ce11-Excludable.v2.rds",
#                 "excluderanges/ce10.Boyle.ce10-Excludable.v2.rds",
#                 "excluderanges/ce10.Kundaje.ce10-Excludable.rds",
#                 "excluderanges/danRer10.Domingues.Excludableed.rds",
#                 "excluderanges/danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome.rds",
#                 "excluderanges/dm6.Boyle.dm6-Excludable.v2.rds",
#                 "excluderanges/dm3.Boyle.dm3-Excludable.v2.rds",
#                 "excluderanges/dm3.Kundaje.dm3-Excludable.rds",
#                 "excluderanges/TAIR10.Wimberley.predicted_excluded_list_sorted_0.6.rds",
#                 "excluderanges/TAIR10.Klasfeld.arabidopsis_Excludable_20inputs.rds",
#                 "excluderanges/TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs.rds",
#                 "excluderanges/T2T.Lareau.chm13v2.0_peaks.rds",
#                 "excluderanges/hg38.Lareau.hg38_peaks.rds",
#                 "excluderanges/hg19.Lareau.hg19_peaks.rds",
#                 "excluderanges/mm10.Lareau.mm10_peaks.rds",
#                 "excluderanges/mm9.Lareau.mm9_peaks.rds",
#                 "excluderanges/hg19.UCSC.numtS.rds",
#                 "excluderanges/mm9.UCSC.numtS.rds",
#                 "excluderanges/T2T.CHM13.chm13.draft_v2.0.cen_mask.rds",
#                 "excluderanges/T2T.CHM13.chm13.draft_v1.1.telomere.rds",
#                 "excluderanges/T2T.UCSC.censat.rds",
#                 "excluderanges/T2T.UCSC.gap.rds",
#                 "excluderanges/T2T.UCSC.hgUnique.hg38.rds",
#                 "excluderanges/hg38.UCSC.centromere.rds",
#                 "excluderanges/hg38.UCSC.telomere.rds",
#                 "excluderanges/hg38.UCSC.short_arm.rds",
#                 "excluderanges/hg38.UCSC.heterochromatin.rds",
#                 "excluderanges/hg38.UCSC.contig.rds",
#                 "excluderanges/hg38.UCSC.scaffold.rds",
#                 "excluderanges/hg19.UCSC.centromere.rds",
#                 "excluderanges/hg19.UCSC.telomere.rds",
#                 "excluderanges/hg19.UCSC.short_arm.rds",
#                 "excluderanges/hg19.UCSC.heterochromatin.rds",
#                 "excluderanges/hg19.UCSC.clone.rds",
#                 "excluderanges/hg19.UCSC.contig.rds",
#                 "excluderanges/mm39.UCSC.centromere.rds",
#                 "excluderanges/mm39.UCSC.telomere.rds",
#                 "excluderanges/mm39.UCSC.short_arm.rds",
#                 "excluderanges/mm39.UCSC.contig.rds",
#                 "excluderanges/mm39.UCSC.scaffold.rds",
#                 "excluderanges/mm10.UCSC.centromere.rds",
#                 "excluderanges/mm10.UCSC.telomere.rds",
#                 "excluderanges/mm10.UCSC.short_arm.rds",
#                 "excluderanges/mm10.UCSC.clone.rds",
#                 "excluderanges/mm10.UCSC.contig.rds",
#                 "excluderanges/mm10.UCSC.other.rds",
#                 "excluderanges/mm9.UCSC.centromere.rds",
#                 "excluderanges/mm9.UCSC.fragment.rds",
#                 "excluderanges/mm9.UCSC.contig.rds",
#                 "excluderanges/danRer10.UCSC.contig.rds",
#                 "excluderanges/danRer10.UCSC.scaffold.rds",
#                 "excluderanges/dm6.UCSC.other.rds",
#                 "excluderanges/dm3.UCSC.contig.rds",
#                 "excluderanges/dm3.UCSC.scaffold.rds",
#                 "excluderanges/TAIR10.UCSC.araTha1.gap.rds"),
#   Tags = c(paste("ce10;Anshul Kundaje;Stanford;exclusion;blacklisted"),
#            paste("dm3;Anshul Kundaje;Stanford;exclusion;blacklisted"),
#            paste("hg19;Bradley Bernstein;Broad;exclusion;blacklisted"),
#            paste("hg19;Ewan Birney;EBI;exclusion;blacklisted"),
#            paste("hg19;Gregory Crawford;Duke;exclusion;blacklisted"),
#            paste("hg19;Barbara Wold;Caltech;exclusion;blacklisted"),
#            paste("hg19;Gene Yeo;UCSD;exclusion;blacklisted"),
#            paste("hg38;Bradley Bernstein;Broad;exclusion;blacklisted"),
#            paste("hg38;Anshul Kundaje;Stanford;exclusion;blacklisted"),
#            paste("hg38;Anshul Kundaje;Stanford;exclusion;blacklisted"),
#            paste("hg38;Tim Reddy;Duke;exclusion;blacklisted"),
#            paste("hg38;Barbara Wold;Caltech;exclusion;blacklisted"),
#            paste("hg38;Gene Yeo;UCSD;exclusion;blacklisted"),
#            paste("mm10;Ross Hardison;PennState;exclusion;blacklisted"),
#            paste("mm10;Ross Hardison;PennState;exclusion;blacklisted"),
#            paste("mm10;Anshul Kundaje;Stanford;exclusion;blacklisted"),
#            paste("mm10;Anshul Kundaje;Stanford;exclusion;blacklisted"),
#            paste("mm10;Barbara Wold;Caltech;exclusion;blacklisted"),
#            paste("mm9;Barbara Wold;Caltech;exclusion;blacklisted"),
#            "hg19;centromere;UCSC", 
#            "hg19;clone;UCSC", 
#            "hg19;contig;UCSC", 
#            "hg19;heterochromatin;UCSC", 
#            "hg19;scaffold;UCSC", 
#            "hg19;short_arm;UCSC", 
#            "hg19;telomere;UCSC", 
#            "hg38;contig;UCSC", 
#            "hg38;heterochromatin;UCSC", 
#            "hg38;scaffold;UCSC", 
#            "hg38;short_arm;UCSC", 
#            "hg38;telomere;UCSC", 
#            "mm10;centromere;UCSC", 
#            "mm10;clone;UCSC", 
#            "mm10;contig;UCSC", 
#            "mm10;fragment;UCSC", 
#            "mm10;other;UCSC", 
#            "mm10;short_arm;UCSC", 
#            "mm10;telomere;UCSC", 
#            "mm9;centromere;UCSC", 
#            "mm9;contig;UCSC", 
#            "mm9;fragment;UCSC",
#            "mm10;scaffold;UCSC",
#            "T2T;excluderanges",
#            "hg38;Kundaje;excluderanges",
#            "hg38;Bernstein;excluderanges",
#            "hg38;Boyle;excluderanges",
#            "hg38;Kundaje;excluderanges",
#            "hg38;Lareau;excluderanges",
#            "hg38;Reddy;excluderanges",
#            "hg38;Wimberley;excluderanges",
#            "hg38;Wold;excluderanges",
#            "hg38;Yeo;excluderanges",
#            "hg19;Boyle;hg19-excluderanges",
#            "hg19;Bernstein;excluderanges",
#            "hg19;Birney;excluderanges",
#            "hg19;Crawford;excluderanges",
#            "hg19;Lareau;excluderanges",
#            "hg19;Wold;excluderanges",
#            "hg19;Yeo;excluderanges",
#            "mm39;excluderanges",
#            "mm10;Boyle;excluderanges",
#            "mm10;Hardison;excluderanges",
#            "mm10;Hardison;excluderanges",
#            "mm10;Kundaje;excluderanges",
#            "mm10;Kundaje;excluderanges",
#            "mm10;Lareau;excluderanges",
#            "mm10;Wold;excluderanges",
#            "mm9;Lareau;excluderanges",
#            "mm9;Wold;excluderanges",
#            "ce11;Boyle;excluderanges",
#            "ce10;Boyle;excluderanges",
#            "ce10;Kundaje;excluderanges",
#            "danRer10;Domingues;excluderanges",
#            "danRer10;Yang;excluderanges",
#            "dm6;Boyle;excluderanges",
#            "dm3;Boyle;excluderanges",
#            "dm3;Kundaje;excluderanges",
#            "TAIR10;Wimberley;excluderanges",
#            "TAIR10;Klasfeld;aexcluderanges",
#            "TAIR10;Klasfeld;excluderanges",
#            "T2T;Lareau;NUMTs",
#            "hg38;Lareau;NUMTs",
#            "hg19;Lareau;NUMTs",
#            "mm10;Lareau;NUMTs",
#            "mm9;Lareau;NUMTs",
#            "hg19;UCSC;NUMTs",
#            "mm9;UCSC;NUMTs",
#            "T2T;CHM13;centromere",
#            "T2T;CHM13;telomere",
#            "T2T;UCSC;censat",
#            "T2T;UCSC;gap",
#            "T2T;UCSC;hgUnique",
#            "hg38;UCSC;centromere",
#            "hg38;UCSC;telomere",
#            "hg38;UCSC;short_arm",
#            "hg38;UCSC;heterochromatin",
#            "hg38;UCSC;contig",
#            "hg38;UCSC;scaffold",
#            "hg19;UCSC;centromere",
#            "hg19;UCSC;telomere",
#            "hg19;UCSC;short_arm",
#            "hg19;UCSC;heterochromatin",
#            "hg19;UCSC;clone",
#            "hg19;UCSC;contig",
#            "mm39;UCSC;centromere",
#            "mm39;UCSC;telomere",
#            "mm39;UCSC;short_arm",
#            "mm39;UCSC;contig",
#            "mm39;UCSC;scaffold",
#            "mm10;UCSC;centromere",
#            "mm10;UCSC;telomere",
#            "mm10;UCSC;short_arm",
#            "mm10;UCSC;clone",
#            "mm10;UCSC;contig",
#            "mm10;UCSC;other",
#            "mm9;UCSC;centromere",
#            "mm9;UCSC;fragment",
#            "mm9;UCSC;contig",
#            "danRer10;UCSC;contig",
#            "danRer10;UCSC;scaffold",
#            "dm6;UCSC;other",
#            "dm3;UCSC;contig",
#            "dm3;UCSC;scaffold",
#            "TAIR10;UCSC;gap"),
#   Notes = c(rep("", 42),
#             "Original region counts: 2066; Filtered for standard chromosomes: 2066",
#             "Original region counts: 910; Filtered for standard chromosomes: 910",
#             "Original region counts: 12052; Filtered for standard chromosomes: 12052",
#             "Original region counts: 636; Filtered for standard chromosomes: 636",
#             "Original region counts: 38; Filtered for standard chromosomes: 38",
#             "Original region counts: 820; Filtered for standard chromosomes: 820",
#             "Original region counts: 401; Filtered for standard chromosomes: 396",
#             "Original region counts: 5078; Filtered for standard chromosomes: 5078",
#             "Original region counts: 299; Filtered for standard chromosomes: 299",
#             "Original region counts: 56; Filtered for standard chromosomes: 56",
#             "Original region counts: 834; Filtered for standard chromosomes: 834",
#             "Original region counts: 9035; Filtered for standard chromosomes: 9035",
#             "Original region counts: 411; Filtered for standard chromosomes: 411",
#             "Original region counts: 1649; Filtered for standard chromosomes: 1566",
#             "Original region counts: 902; Filtered for standard chromosomes: 902",
#             "Original region counts: 295; Filtered for standard chromosomes: 295",
#             "Original region counts: 57; Filtered for standard chromosomes: 57",
#             "Original region counts: 3147; Filtered for standard chromosomes: 3147",
#             "Original region counts: 3435; Filtered for standard chromosomes: 3435",
#             "Original region counts: 7865; Filtered for standard chromosomes: 7865",
#             "Original region counts: 5552; Filtered for standard chromosomes: 5552",
#             "Original region counts: 3010; Filtered for standard chromosomes: 3010",
#             "Original region counts: 164; Filtered for standard chromosomes: 164",
#             "Original region counts: 523; Filtered for standard chromosomes: 523",
#             "Original region counts: 123; Filtered for standard chromosomes: 123",
#             "Original region counts: 3415; Filtered for standard chromosomes: 3415",
#             "Original region counts: 123; Filtered for standard chromosomes: 123",
#             "Original region counts: 97; Filtered for standard chromosomes: 97",
#             "Original region counts: 100; Filtered for standard chromosomes: 100",
#             "Original region counts: 122; Filtered for standard chromosomes: 122",
#             "Original region counts: 62; Filtered for standard chromosomes: 57",
#             "Original region counts: 853; Filtered for standard chromosomes: 853",
#             "Original region counts: 182; Filtered for standard chromosomes: 182",
#             "Original region counts: 271; Filtered for standard chromosomes: 248",
#             "Original region counts: 492; Filtered for standard chromosomes: 306",
#             "Original region counts: 887; Filtered for standard chromosomes: 887",
#             "Original region counts: 83; Filtered for standard chromosomes: 83",
#             "Original region counts: 36; Filtered for standard chromosomes: 36",
#             "Original region counts: 817; Filtered for standard chromosomes: 817",
#             "Original region counts: 784; Filtered for standard chromosomes: 784",
#             "Original region counts: 779; Filtered for standard chromosomes: 779",
#             "Original region counts: 387; Filtered for standard chromosomes: 387",
#             "Original region counts: 395; Filtered for standard chromosomes: 395",
#             "Original region counts: 766; Filtered for standard chromosomes: 766",
#             "Original region counts: 172; Filtered for standard chromosomes: 172",
#             "Original region counts: 23; Filtered for standard chromosomes: 23",
#             "Original region counts: 48; Filtered for standard chromosomes: 48",
#             "Original region counts: 2523; Filtered for standard chromosomes: 2523",
#             "Original region counts: 5; Filtered for standard chromosomes: 5",
#             "Original region counts: 615; Filtered for standard chromosomes: 615",
#             "Original region counts: 109; Filtered for standard chromosomes: 109",
#             "Original region counts: 48; Filtered for standard chromosomes: 48",
#             "Original region counts: 5; Filtered for standard chromosomes: 5",
#             "Original region counts: 11; Filtered for standard chromosomes: 11",
#             "Original region counts: 285; Filtered for standard chromosomes: 285",
#             "Original region counts: 478; Filtered for standard chromosomes: 254",
#             "Original region counts: 24; Filtered for standard chromosomes: 24",
#             "Original region counts: 46; Filtered for standard chromosomes: 46",
#             "Original region counts: 5; Filtered for standard chromosomes: 5",
#             "Original region counts: 12; Filtered for standard chromosomes: 12",
#             "Original region counts: 207; Filtered for standard chromosomes: 107",
#             "Original region counts: 163; Filtered for standard chromosomes: 163",
#             "Original region counts: 20; Filtered for standard chromosomes: 20",
#             "Original region counts: 42; Filtered for standard chromosomes: 42",
#             "Original region counts: 21; Filtered for standard chromosomes: 21",
#             "Original region counts: 60; Filtered for standard chromosomes: 60",
#             "Original region counts: 181; Filtered for standard chromosomes: 115",
#             "Original region counts: 20; Filtered for standard chromosomes: 20",
#             "Original region counts: 42; Filtered for standard chromosomes: 42",
#             "Original region counts: 21; Filtered for standard chromosomes: 21",
#             "Original region counts: 114; Filtered for standard chromosomes: 4",
#             "Original region counts: 104; Filtered for standard chromosomes: 104",
#             "Original region counts: 384; Filtered for standard chromosomes: 383",
#             "Original region counts: 21; Filtered for standard chromosomes: 21",
#             "Original region counts: 709; Filtered for standard chromosomes: 436",
#             "Original region counts: 281; Filtered for standard chromosomes: 105",
#             "Original region counts: 2338; Filtered for standard chromosomes: 2338",
#             "Original region counts: 18955; Filtered for standard chromosomes: 16496",
#             "Original region counts: 572; Filtered for standard chromosomes: 268",
#             "Original region counts: 37665; Filtered for standard chromosomes: 7",
#             "Original region counts: 8; Filtered for standard chromosomes: 1",
#             "Original region counts: 357; Filtered for standard chromosomes: 357"))



# NEW objects only
meta <- data.frame(
  Title = c("T2T.excluderanges",
            "hg38.Kundaje.GRCh38_unified_Excludable",
            "hg38.Bernstein.Mint_Excludable_GRCh38",
            "hg38.Boyle.hg38-Excludable.v2",
            "hg38.Kundaje.GRCh38.Excludable",
            "hg38.Lareau.hg38.full.Excludable",
            "hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38",
            "hg38.Wimberley.peakPass60Perc_sorted",
            "hg38.Wold.hg38mitoExcludable",
            "hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed",
            "hg19.Boyle.hg19-Excludable.v2",
            "hg19.Bernstein.Mint_Excludable_hg19",
            "hg19.Birney.wgEncodeDacMapabilityConsensusExcludable",
            "hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable",
            "hg19.Lareau.hg19.full.Excludable",
            "hg19.Wold.hg19mitoExcludable",
            "hg19.Yeo.eCLIP_Excludableregions.hg19",
            "mm39.excluderanges",
            "mm10.Boyle.mm10-Excludable.v2",
            "mm10.Hardison.Excludable.full",
            "mm10.Hardison.psuExcludable.mm10",
            "mm10.Kundaje.anshul.Excludable.mm10",
            "mm10.Kundaje.mm10.Excludable",
            "mm10.Lareau.mm10.full.Excludable",
            "mm10.Wold.mm10mitoExcludable",
            "mm9.Lareau.mm9.full.Excludable",
            "mm9.Wold.mm9mitoExcludable",
            "ce11.Boyle.ce11-Excludable.v2",
            "ce10.Boyle.ce10-Excludable.v2",
            "ce10.Kundaje.ce10-Excludable",
            "danRer10.Domingues.Excludableed",
            "danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome",
            "dm6.Boyle.dm6-Excludable.v2",
            "dm3.Boyle.dm3-Excludable.v2",
            "dm3.Kundaje.dm3-Excludable",
            "TAIR10.Wimberley.predicted_excluded_list_sorted_0.6",
            "TAIR10.Klasfeld.arabidopsis_Excludable_20inputs",
            "TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs",
            "T2T.Lareau.chm13v2.0_peaks",
            "hg38.Lareau.hg38_peaks",
            "hg19.Lareau.hg19_peaks",
            "mm10.Lareau.mm10_peaks",
            "mm9.Lareau.mm9_peaks",
            "hg19.UCSC.numtS",
            "mm9.UCSC.numtS",
            "T2T.CHM13.chm13.draft_v2.0.cen_mask",
            "T2T.CHM13.chm13.draft_v1.1.telomere",
            "T2T.UCSC.censat",
            "T2T.UCSC.gap",
            "T2T.UCSC.hgUnique.hg38",
            "hg38.UCSC.centromere",
            "hg38.UCSC.telomere",
            "hg38.UCSC.short_arm",
            "hg38.UCSC.heterochromatin",
            "hg38.UCSC.contig",
            "hg38.UCSC.scaffold",
            "hg19.UCSC.centromere",
            "hg19.UCSC.telomere",
            "hg19.UCSC.short_arm",
            "hg19.UCSC.heterochromatin",
            "hg19.UCSC.clone",
            "hg19.UCSC.contig",
            "mm39.UCSC.centromere",
            "mm39.UCSC.telomere",
            "mm39.UCSC.short_arm",
            "mm39.UCSC.contig",
            "mm39.UCSC.scaffold",
            "mm10.UCSC.centromere",
            "mm10.UCSC.telomere",
            "mm10.UCSC.short_arm",
            "mm10.UCSC.clone",
            "mm10.UCSC.contig",
            "mm10.UCSC.other",
            "mm9.UCSC.centromere",
            "mm9.UCSC.fragment",
            "mm9.UCSC.contig",
            "danRer10.UCSC.contig",
            "danRer10.UCSC.scaffold",
            "dm6.UCSC.other",
            "dm3.UCSC.contig",
            "dm3.UCSC.scaffold",
            "TAIR10.UCSC.araTha1.gap",
            "hg38.GreyListChIP.STAR_36bp_1000merge",
            "hg38.GreyListChIP.STAR_101bp_1000merge",
            "T2T.GreyListChIP.STAR_36bp_1000merge",
            "T2T.GreyListChIP.STAR_101bp_1000merge",
            "mm10.GreyListChIP.STAR_36bp_1000merge",
            "mm10.GreyListChIP.STAR_50bp_1000merge",
            "mm39.GreyListChIP.STAR_36bp_1000merge",
            "mm39.GreyListChIP.STAR_50bp_1000merge"
  ),
  Description = c("Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined as a combination of hg38.Lareau.hg38_peaks, hg38.Boyle.hg38-Excludable.v2, and hg38.Wimberley.peakPass60Perc_sorted, followed by manual curation. Supersedes hg38.Kundaje.GRCh38.Excludable.",
                  "Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Defined by the ewimberley/peakPass software",
                  "Definition method unknown",
                  "Defined from eCLIP data",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined from Mint-ChIP (low input, multiplexed ChIP-seq) data",
                  "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), centromeric repeats, chrM, High/Low mappability islands",
                  "Defined by the ENCODE consortium, includes satellite repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha, BSR/Beta), human satellite repeat HSATII, chrM, ribosomal subunit consensus sequences LSU-rRNA_Hsa, SSU-rRNA_Hsa. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Definition method unknown",
                  "Defined from eCLIP data, includes skyscraper, rRNA pseudogene, unreliably mapped satellite repeat, and low complexity skyscraper peak regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Definition method unknown",
                  "Definition method unknown",
                  "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
                  "Defined by Anshul Kundaje as a part of ENCODE and modENCODE consortia",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Definition method unknown",
                  "ENCODE excludable regions combined with regions of high homology to mtDNA (NUMT regions)",
                  "Definition method unknown",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by Anshul Kundaje, superseded by ce10.Boyle.ce10-Excludable.v2",
                  "Defined manually using total RNA-seq.",
                  "Defined via MACS2 peak calling using ChIP-seq (PMID: 33239788)",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions",
                  "Defined by Anshul Kundaje. Contains heterochromatin chromosomes chr2LHet. Superseded by dm3.Boyle.dm3-Excludable.v2",
                  "Defined by the ewimberley/peakPass software",
                  "Defined by the Boyle-Lab/Blacklist software, High Signal and Low Mappability regions (DOI: 10.1101/2022.02.27.482177)",
                  "Defined by the green screen pipeline (DOI: 10.1101/2022.02.27.482177)",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Regions of high homology to mtDNA (NUMT regions) defined by caleblareau/mitoblacklist",
                  "Human NumtS mitochondrial sequence",
                  "Mouse NumtS mitochondrial sequence",
                  "Centromeric satellite masking bed file (v2.0)",
                  "Telomere identified by the VGP pipeline (v1.1)",
                  "T2T peri/centromeric satellite annotation (v2.0, 20220329, CHM13 v2.0)",
                  "Locations of assembly gaps, as determine by strings of 'N' characters (v1.0)",
                  "Regions unique to the T2T-CHM13 v2.0 assembly compared to the GRCh38/hg38 and GRCh37/hg19 reference assemblies",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps from large blocks of heterochromatin",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps from large blocks of heterochromatin",
                  "Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Gaps from centromeres",
                  "Gaps from telomeres",
                  "Gaps on the short arm of the chromosome",
                  "Gaps between clones in the same map contig. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds",
                  "Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps from centromeres",
                  "Gaps between the contigs of a draft clone. (In this context, a contig is a set of overlapping sequence reads). Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds. Has extra chromosomes, use keepStandardChromosomes() filtering",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Sequence of Ns in the assembly that were not marked as gaps in the AGP (A Golden Path) assembly definition file",
                  "Gaps between contigs in scaffolds",
                  "Gaps between scaffolds in chromosome assemblies",
                  "Gaps in the May 2011 Arabidopsis thaliana genome assembly",
                  
                  "GreyListChIP-generated exclusion set using STAR-aligned 36bp single-end reads (381 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 100bp and 101bp paired-end reads (274 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 36bp single-end reads (381 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 101bp single-end reads (381 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 36bp single-end reads (106 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 50bp single-end reads (175 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 36bp single-end reads (106 files combined), merging regions within 1,000bp",
                  "GreyListChIP-generated exclusion set using STAR-aligned 50bp single-end reads (176 files combined), merging regions within 1,000bp"),
  BiocVersion = c(rep("3.16", 82), rep("3.20", 8)),
  Genome = c("T2T-CHM13",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "mm39",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm9",
             "mm9",
             "ce11",
             "ce10",
             "ce10",
             "danRer10",
             "danRer10",
             "dm6",
             "dm3",
             "dm3",
             "TAIR10",
             "TAIR10",
             "TAIR10",
             "T2T-CHM13",
             "hg38",
             "hg19",
             "mm10",
             "mm9",
             "hg19",
             "mm9",
             "T2T-CHM13",
             "T2T-CHM13",
             "T2T-CHM13",
             "T2T-CHM13",
             "T2T-CHM13",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg38",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "hg19",
             "mm39",
             "mm39",
             "mm39",
             "mm39",
             "mm39",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm10",
             "mm9",
             "mm9",
             "mm9",
             "danRer10",
             "danRer10",
             "dm6",
             "dm3",
             "dm3",
             "TAIR10",
             
             "hg38",
             "hg38",
             "T2T",
             "T2T",
             "mm10",
             "mm10",
             "mm39",
             "mm39"),
  SourceType = rep("RDS", 90),
  SourceUrl = c(rep("https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2kfHOWFBSJT?usp=sharing", 82), rep("https://drive.google.com/drive/folders/171xy6RypzjdrSfltbbEmDUKR6X7A-_NR?usp=sharing",8)),
  SourceVersion = c(rep("September 22 2022", 82), rep("November 25 2024", 8)),
  Species = c("Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Caenorhabditis elegans",
              "Caenorhabditis elegans",
              "Caenorhabditis elegans",
              "Danio rerio",
              "Danio rerio",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Arabidopsis thaliana",
              "Arabidopsis thaliana",
              "Arabidopsis thaliana",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Homo sapiens",
              "Mus musculus",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Danio rerio",
              "Danio rerio",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Drosophila melanogaster",
              "Arabidopsis thaliana",
              
              "Homo sapiens",
              "Homo sapiens", 
              "Homo sapiens",
              "Homo sapiens",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus",
              "Mus musculus"
  ),
  TaxonomyId = c("9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "6239",
                 "6239",
                 "6239",
                 "7955",
                 "7955",
                 "7227",
                 "7227",
                 "7227",
                 "3702",
                 "3702",
                 "3702",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "9606",
                 "10090",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "10090",
                 "7955",
                 "7955",
                 "7227",
                 "7227",
                 "7227",
                 "3702",
                 
                 "9606",
                 "9606",
                 "9606",
                 "9606",
                 "10090",
                 "10090",
                 "10090",
                 "10090"),
  Coordinate_1_based = c(TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         FALSE,
                         FALSE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         FALSE,
                         TRUE,
                         
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE,
                         TRUE),
  DataProvider = c("excluderanges",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "excluderanges",
                   "GitHub",
                   "ENCODE",
                   "ENCODE",
                   "ENCODE",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "ENCODE",
                   "GitHub",
                   "GitHub",
                   "Stanford.edu",
                   "GitHub",
                   "Publication",
                   "GitHub",
                   "GitHub",
                   "Stanford.edu",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "GitHub",
                   "UCSC",
                   "UCSC",
                   "CHM13",
                   "CHM13",
                   "UCSChub",
                   "UCSChub",
                   "UCSChub",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSC",
                   "UCSChub",
                   
                   "GreyListChIP",
                   "GreyListChIP",
                   "GreyListChIP",
                   "GreyListChIP",
                   "GreyListChIP",
                   "GreyListChIP",
                   "GreyListChIP",
                   "GreyListChIP"),
  Maintainer = rep("Mikhail Dozmorov <mikhail.dozmorov@gmail.com>", 90),
  RDataClass = rep("GRanges", 90),
  DispatchClass = rep("RDS", 90),
  RDataPath = c("excluderanges/T2T.excluderanges.rds",
                "excluderanges/hg38.Kundaje.GRCh38_unified_Excludable.rds",
                "excluderanges/hg38.Bernstein.Mint_Excludable_GRCh38.rds",
                "excluderanges/hg38.Boyle.hg38-Excludable.v2.rds",
                "excluderanges/hg38.Kundaje.GRCh38.Excludable.rds",
                "excluderanges/hg38.Lareau.hg38.full.Excludable.rds",
                "excluderanges/hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds",
                "excluderanges/hg38.Wimberley.peakPass60Perc_sorted.rds",
                "excluderanges/hg38.Wold.hg38mitoExcludable.rds",
                "excluderanges/hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds",
                "excluderanges/hg19.Boyle.hg19-Excludable.v2.rds",
                "excluderanges/hg19.Bernstein.Mint_Excludable_hg19.rds",
                "excluderanges/hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds",
                "excluderanges/hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds",
                "excluderanges/hg19.Lareau.hg19.full.Excludable.rds",
                "excluderanges/hg19.Wold.hg19mitoExcludable.rds",
                "excluderanges/hg19.Yeo.eCLIP_Excludableregions.hg19.rds",
                "excluderanges/mm39.excluderanges.rds",
                "excluderanges/mm10.Boyle.mm10-Excludable.v2.rds",
                "excluderanges/mm10.Hardison.Excludable.full.rds",
                "excluderanges/mm10.Hardison.psuExcludable.mm10.rds",
                "excluderanges/mm10.Kundaje.anshul.Excludable.mm10.rds",
                "excluderanges/mm10.Kundaje.mm10.Excludable.rds",
                "excluderanges/mm10.Lareau.mm10.full.Excludable.rds",
                "excluderanges/mm10.Wold.mm10mitoExcludable.rds",
                "excluderanges/mm9.Lareau.mm9.full.Excludable.rds",
                "excluderanges/mm9.Wold.mm9mitoExcludable.rds",
                "excluderanges/ce11.Boyle.ce11-Excludable.v2.rds",
                "excluderanges/ce10.Boyle.ce10-Excludable.v2.rds",
                "excluderanges/ce10.Kundaje.ce10-Excludable.rds",
                "excluderanges/danRer10.Domingues.Excludableed.rds",
                "excluderanges/danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome.rds",
                "excluderanges/dm6.Boyle.dm6-Excludable.v2.rds",
                "excluderanges/dm3.Boyle.dm3-Excludable.v2.rds",
                "excluderanges/dm3.Kundaje.dm3-Excludable.rds",
                "excluderanges/TAIR10.Wimberley.predicted_excluded_list_sorted_0.6.rds",
                "excluderanges/TAIR10.Klasfeld.arabidopsis_Excludable_20inputs.rds",
                "excluderanges/TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs.rds",
                "excluderanges/T2T.Lareau.chm13v2.0_peaks.rds",
                "excluderanges/hg38.Lareau.hg38_peaks.rds",
                "excluderanges/hg19.Lareau.hg19_peaks.rds",
                "excluderanges/mm10.Lareau.mm10_peaks.rds",
                "excluderanges/mm9.Lareau.mm9_peaks.rds",
                "excluderanges/hg19.UCSC.numtS.rds",
                "excluderanges/mm9.UCSC.numtS.rds",
                "excluderanges/T2T.CHM13.chm13.draft_v2.0.cen_mask.rds",
                "excluderanges/T2T.CHM13.chm13.draft_v1.1.telomere.rds",
                "excluderanges/T2T.UCSC.censat.rds",
                "excluderanges/T2T.UCSC.gap.rds",
                "excluderanges/T2T.UCSC.hgUnique.hg38.rds",
                "excluderanges/hg38.UCSC.centromere.rds",
                "excluderanges/hg38.UCSC.telomere.rds",
                "excluderanges/hg38.UCSC.short_arm.rds",
                "excluderanges/hg38.UCSC.heterochromatin.rds",
                "excluderanges/hg38.UCSC.contig.rds",
                "excluderanges/hg38.UCSC.scaffold.rds",
                "excluderanges/hg19.UCSC.centromere.rds",
                "excluderanges/hg19.UCSC.telomere.rds",
                "excluderanges/hg19.UCSC.short_arm.rds",
                "excluderanges/hg19.UCSC.heterochromatin.rds",
                "excluderanges/hg19.UCSC.clone.rds",
                "excluderanges/hg19.UCSC.contig.rds",
                "excluderanges/mm39.UCSC.centromere.rds",
                "excluderanges/mm39.UCSC.telomere.rds",
                "excluderanges/mm39.UCSC.short_arm.rds",
                "excluderanges/mm39.UCSC.contig.rds",
                "excluderanges/mm39.UCSC.scaffold.rds",
                "excluderanges/mm10.UCSC.centromere.rds",
                "excluderanges/mm10.UCSC.telomere.rds",
                "excluderanges/mm10.UCSC.short_arm.rds",
                "excluderanges/mm10.UCSC.clone.rds",
                "excluderanges/mm10.UCSC.contig.rds",
                "excluderanges/mm10.UCSC.other.rds",
                "excluderanges/mm9.UCSC.centromere.rds",
                "excluderanges/mm9.UCSC.fragment.rds",
                "excluderanges/mm9.UCSC.contig.rds",
                "excluderanges/danRer10.UCSC.contig.rds",
                "excluderanges/danRer10.UCSC.scaffold.rds",
                "excluderanges/dm6.UCSC.other.rds",
                "excluderanges/dm3.UCSC.contig.rds",
                "excluderanges/dm3.UCSC.scaffold.rds",
                "excluderanges/TAIR10.UCSC.araTha1.gap.rds",
                
                "excluderanges/hg38.GreyListChIP.STAR_36bp_1000merge.rds",
                "excluderanges/hg38.GreyListChIP.STAR_101bp_1000merge.rds",
                "excluderanges/T2T.GreyListChIP.STAR_36bp_1000merge.rds",
                "excluderanges/star_k101_1000_grey_T2T.rds",
                "excluderanges/mm10.GreyListChIP.STAR_36bp_1000merge.rds",
                "excluderanges/mm10.GreyListChIP.STAR_50bp_1000merge.rds",
                "excluderanges/mm39.GreyListChIP.STAR_36bp_1000merge.rds",
                "excluderanges/mm39.GreyListChIP.STAR_50bp_1000merge.rds"),
  Tags = c("T2T;excluderanges",
           "hg38;Kundaje;excluderanges",
           "hg38;Bernstein;excluderanges",
           "hg38;Boyle;excluderanges",
           "hg38;Kundaje;excluderanges",
           "hg38;Lareau;excluderanges",
           "hg38;Reddy;excluderanges",
           "hg38;Wimberley;excluderanges",
           "hg38;Wold;excluderanges",
           "hg38;Yeo;excluderanges",
           "hg19;Boyle;hg19-excluderanges",
           "hg19;Bernstein;excluderanges",
           "hg19;Birney;excluderanges",
           "hg19;Crawford;excluderanges",
           "hg19;Lareau;excluderanges",
           "hg19;Wold;excluderanges",
           "hg19;Yeo;excluderanges",
           "mm39;excluderanges",
           "mm10;Boyle;excluderanges",
           "mm10;Hardison;excluderanges",
           "mm10;Hardison;excluderanges",
           "mm10;Kundaje;excluderanges",
           "mm10;Kundaje;excluderanges",
           "mm10;Lareau;excluderanges",
           "mm10;Wold;excluderanges",
           "mm9;Lareau;excluderanges",
           "mm9;Wold;excluderanges",
           "ce11;Boyle;excluderanges",
           "ce10;Boyle;excluderanges",
           "ce10;Kundaje;excluderanges",
           "danRer10;Domingues;excluderanges",
           "danRer10;Yang;excluderanges",
           "dm6;Boyle;excluderanges",
           "dm3;Boyle;excluderanges",
           "dm3;Kundaje;excluderanges",
           "TAIR10;Wimberley;excluderanges",
           "TAIR10;Klasfeld;aexcluderanges",
           "TAIR10;Klasfeld;excluderanges",
           "T2T;Lareau;NUMTs",
           "hg38;Lareau;NUMTs",
           "hg19;Lareau;NUMTs",
           "mm10;Lareau;NUMTs",
           "mm9;Lareau;NUMTs",
           "hg19;UCSC;NUMTs",
           "mm9;UCSC;NUMTs",
           "T2T;CHM13;centromere",
           "T2T;CHM13;telomere",
           "T2T;UCSC;censat",
           "T2T;UCSC;gap",
           "T2T;UCSC;hgUnique",
           "hg38;UCSC;centromere",
           "hg38;UCSC;telomere",
           "hg38;UCSC;short_arm",
           "hg38;UCSC;heterochromatin",
           "hg38;UCSC;contig",
           "hg38;UCSC;scaffold",
           "hg19;UCSC;centromere",
           "hg19;UCSC;telomere",
           "hg19;UCSC;short_arm",
           "hg19;UCSC;heterochromatin",
           "hg19;UCSC;clone",
           "hg19;UCSC;contig",
           "mm39;UCSC;centromere",
           "mm39;UCSC;telomere",
           "mm39;UCSC;short_arm",
           "mm39;UCSC;contig",
           "mm39;UCSC;scaffold",
           "mm10;UCSC;centromere",
           "mm10;UCSC;telomere",
           "mm10;UCSC;short_arm",
           "mm10;UCSC;clone",
           "mm10;UCSC;contig",
           "mm10;UCSC;other",
           "mm9;UCSC;centromere",
           "mm9;UCSC;fragment",
           "mm9;UCSC;contig",
           "danRer10;UCSC;contig",
           "danRer10;UCSC;scaffold",
           "dm6;UCSC;other",
           "dm3;UCSC;contig",
           "dm3;UCSC;scaffold",
           "TAIR10;UCSC;gap",
           
           "hg38;GreyListChIP;STAR_36bp",
           "hg38;GreyListChIP;STAR_101bp",
           "T2T;GreyListChIP;STAR_36bp",
           "T2T;GreyListChIP;STAR_101bp",
           "mm10;GreyListChIP;STAR_36bp",
           "mm10;GreyListChIP;STAR_50bp",
           "mm39;GreyListChIP;STAR_36bp",
           "mm39;GreyListChIP;STAR_50bp"),
  Notes = c("Original region counts: 2066; Filtered for standard chromosomes: 2066",
            "Original region counts: 910; Filtered for standard chromosomes: 910",
            "Original region counts: 12052; Filtered for standard chromosomes: 12052",
            "Original region counts: 636; Filtered for standard chromosomes: 636",
            "Original region counts: 38; Filtered for standard chromosomes: 38",
            "Original region counts: 820; Filtered for standard chromosomes: 820",
            "Original region counts: 401; Filtered for standard chromosomes: 396",
            "Original region counts: 5078; Filtered for standard chromosomes: 5078",
            "Original region counts: 299; Filtered for standard chromosomes: 299",
            "Original region counts: 56; Filtered for standard chromosomes: 56",
            "Original region counts: 834; Filtered for standard chromosomes: 834",
            "Original region counts: 9035; Filtered for standard chromosomes: 9035",
            "Original region counts: 411; Filtered for standard chromosomes: 411",
            "Original region counts: 1649; Filtered for standard chromosomes: 1566",
            "Original region counts: 902; Filtered for standard chromosomes: 902",
            "Original region counts: 295; Filtered for standard chromosomes: 295",
            "Original region counts: 57; Filtered for standard chromosomes: 57",
            "Original region counts: 3147; Filtered for standard chromosomes: 3147",
            "Original region counts: 3435; Filtered for standard chromosomes: 3435",
            "Original region counts: 7865; Filtered for standard chromosomes: 7865",
            "Original region counts: 5552; Filtered for standard chromosomes: 5552",
            "Original region counts: 3010; Filtered for standard chromosomes: 3010",
            "Original region counts: 164; Filtered for standard chromosomes: 164",
            "Original region counts: 523; Filtered for standard chromosomes: 523",
            "Original region counts: 123; Filtered for standard chromosomes: 123",
            "Original region counts: 3415; Filtered for standard chromosomes: 3415",
            "Original region counts: 123; Filtered for standard chromosomes: 123",
            "Original region counts: 97; Filtered for standard chromosomes: 97",
            "Original region counts: 100; Filtered for standard chromosomes: 100",
            "Original region counts: 122; Filtered for standard chromosomes: 122",
            "Original region counts: 62; Filtered for standard chromosomes: 57",
            "Original region counts: 853; Filtered for standard chromosomes: 853",
            "Original region counts: 182; Filtered for standard chromosomes: 182",
            "Original region counts: 271; Filtered for standard chromosomes: 248",
            "Original region counts: 492; Filtered for standard chromosomes: 306",
            "Original region counts: 887; Filtered for standard chromosomes: 887",
            "Original region counts: 83; Filtered for standard chromosomes: 83",
            "Original region counts: 36; Filtered for standard chromosomes: 36",
            "Original region counts: 817; Filtered for standard chromosomes: 817",
            "Original region counts: 784; Filtered for standard chromosomes: 784",
            "Original region counts: 779; Filtered for standard chromosomes: 779",
            "Original region counts: 387; Filtered for standard chromosomes: 387",
            "Original region counts: 395; Filtered for standard chromosomes: 395",
            "Original region counts: 766; Filtered for standard chromosomes: 766",
            "Original region counts: 172; Filtered for standard chromosomes: 172",
            "Original region counts: 23; Filtered for standard chromosomes: 23",
            "Original region counts: 48; Filtered for standard chromosomes: 48",
            "Original region counts: 2523; Filtered for standard chromosomes: 2523",
            "Original region counts: 5; Filtered for standard chromosomes: 5",
            "Original region counts: 615; Filtered for standard chromosomes: 615",
            "Original region counts: 109; Filtered for standard chromosomes: 109",
            "Original region counts: 48; Filtered for standard chromosomes: 48",
            "Original region counts: 5; Filtered for standard chromosomes: 5",
            "Original region counts: 11; Filtered for standard chromosomes: 11",
            "Original region counts: 285; Filtered for standard chromosomes: 285",
            "Original region counts: 478; Filtered for standard chromosomes: 254",
            "Original region counts: 24; Filtered for standard chromosomes: 24",
            "Original region counts: 46; Filtered for standard chromosomes: 46",
            "Original region counts: 5; Filtered for standard chromosomes: 5",
            "Original region counts: 12; Filtered for standard chromosomes: 12",
            "Original region counts: 207; Filtered for standard chromosomes: 107",
            "Original region counts: 163; Filtered for standard chromosomes: 163",
            "Original region counts: 20; Filtered for standard chromosomes: 20",
            "Original region counts: 42; Filtered for standard chromosomes: 42",
            "Original region counts: 21; Filtered for standard chromosomes: 21",
            "Original region counts: 60; Filtered for standard chromosomes: 60",
            "Original region counts: 181; Filtered for standard chromosomes: 115",
            "Original region counts: 20; Filtered for standard chromosomes: 20",
            "Original region counts: 42; Filtered for standard chromosomes: 42",
            "Original region counts: 21; Filtered for standard chromosomes: 21",
            "Original region counts: 114; Filtered for standard chromosomes: 4",
            "Original region counts: 104; Filtered for standard chromosomes: 104",
            "Original region counts: 384; Filtered for standard chromosomes: 383",
            "Original region counts: 21; Filtered for standard chromosomes: 21",
            "Original region counts: 709; Filtered for standard chromosomes: 436",
            "Original region counts: 281; Filtered for standard chromosomes: 105",
            "Original region counts: 2338; Filtered for standard chromosomes: 2338",
            "Original region counts: 18955; Filtered for standard chromosomes: 16496",
            "Original region counts: 572; Filtered for standard chromosomes: 268",
            "Original region counts: 37665; Filtered for standard chromosomes: 7",
            "Original region counts: 8; Filtered for standard chromosomes: 1",
            "Original region counts: 357; Filtered for standard chromosomes: 357",
            "Combined files: 381, merging regions within 1,000bp",
            "Combined files: 274, merging regions within 1,000bp",
            "Combined files: 381, merging regions within 1,000bp",
            "Combined files: 381, merging regions within 1,000bp",
            "Combined files: 106, merging regions within 1,000bp",
            "Combined files: 175, merging regions within 1,000bp",
            "Combined files: 106, merging regions within 1,000bp",
            "Combined files: 176, merging regions within 1,000bp"))

write.csv(meta, file="inst/extdata/metadata.csv", row.names=FALSE)
