<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to excluderanges • excluderanges</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to excluderanges">
<meta property="og:description" content="excluderanges">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">excluderanges</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/excluderanges.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/excluderanges/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to excluderanges</h1>
                        <h4 data-toc-skip class="author">Mikhail
Dozmorov</h4>
            <address class="author_afil">
      Virginia Commonwealth
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:mikhail.dozmorov@gmail.com" class="email">mikhail.dozmorov@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">2025-08-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dozmorovlab/excluderanges/blob/HEAD/vignettes/excluderanges.Rmd" class="external-link"><code>vignettes/excluderanges.Rmd</code></a></small>
      <div class="hidden name"><code>excluderanges.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="excluderanges-genomic-ranges-of-problematic-genomic-regions">excluderanges, genomic ranges of problematic genomic regions<a class="anchor" aria-label="anchor" href="#excluderanges-genomic-ranges-of-problematic-genomic-regions"></a>
</h2>
<p>Ogata, Jonathan D, Wancen Mu, Eric S Davis, Bingjie Xue, J Chuck
Harrell, Nathan C Sheffield, Douglas H Phanstiel, Michael I Love, and
Mikhail G Dozmorov. “Excluderanges: Exclusion Sets for T2T-CHM13,
GRCm39, and Other Genome Assemblies.” <em>Bioinformatics</em>, April 3,
2023, PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/37067481/" class="external-link">37067481</a>, DOI: <a href="https://doi.org/10.1093/bioinformatics/btad198" class="external-link">10.1093/bioinformatics/btad198</a></p>
<p><img id="excluderanges_logo_by_Mike_Love" src="../reference/figures/logo.png" align="right" width="125"></p>
<p>Coordinates of problematic genomic regions that should be avoided
when working with genomic data. For human, mouse, and selected model
organisms. Package on Bioconductor: <em><a href="https://bioconductor.org/packages/3.16/excluderanges" class="external-link">excluderanges</a></em>.</p>
<!-- badges: start
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
badges: end -->
<p><strong>New (2022-09-20)</strong> - Exclusion sets for human <a href="http://bedbase.org/#/bedsplash/8329d8c624880308ab51ba05149a737d" class="external-link">T2T-CHM13</a>
and mouse <a href="http://bedbase.org/#/bedsplash/edc716833d4b5ee75c34a0692fc353d5" class="external-link">GRCm39/mm39</a>
genome assemblies are available. Download all data from <a href="https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2kfHOWFBSJT?usp=sharing" class="external-link">Google
Drive</a>.</p>
<p><strong>TL;DR</strong> - For human hg38 genome assembly, <a href="https://twitter.com/anshulkundaje/status/1263546023151992832?s=20" class="external-link">Anshul</a>
and we recommend <a href="https://www.encodeproject.org/files/ENCFF356LFX/" class="external-link">ENCFF356LFX
exclusion list regions</a>. Also available as
<code>hg38.Kundaje.GRCh38_unified_Excludable</code> excluderanges object
(AnnotationHub ID: AH95917)
<!--and [BEDbase.org](http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656)-->.</p>
<p>BED files of exclusion regions are available on the <a href="https://www.encodeproject.org/search/?searchTerm=exclusion+list" class="external-link">ENCODE
project</a> website and scattered across various websites, such as <a href="https://github.com/Boyle-Lab/Blacklist/tree/master/lists" class="external-link">Blacklist</a>
<span class="citation">(Amemiya, Kundaje, and Boyle 2019)</span>, <a href="https://github.com/ewimberley/peakPass/tree/main/excludedlists" class="external-link">Peakpass</a>
<span class="citation">(Wimberley and Heber 2019)</span>, <a href="https://github.com/sklasfeld/GreenscreenProject/tree/main/data" class="external-link">Greenscreen</a>
<span class="citation">(Klasfeld and Wagner 2022)</span>. Human and
mouse genome assemblies have the largest number of exclusion sets
generated by multiple labs. These exclusion sets frequently lack
annotation and curation methods, creating uncertainty what to use. The
purpose of this package is to provide a unified place for informed
retrieval of exclusion regions.</p>
<p>Naming convention:
<code>&lt;genome assembly&gt;.&lt;lab&gt;.&lt;original file name&gt;</code>,
e.g.,
<code>hg19.Birney.wgEncodeDacMapabilityConsensusExcludable</code>.</p>
<p>See <a href="../inst/scripts/make-data.R">make-data.R</a> how we
created excluderanges objects.</p>
<div class="section level3">
<h3 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h3>
<p><a href="https://www.bioconductor.org/install/#install-R" class="external-link">Install</a>
the latest release of R, then get the latest version of Bioconductor by
starting R and entering the commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("BiocManager")</span></span>
<span><span class="co"># BiocManager::install(version = "3.16")</span></span></code></pre></div>
<p>Then, install additional packages using the following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># BiocManager::install("AnnotationHub", update = FALSE) </span></span>
<span><span class="co"># BiocManager::install("GenomicRanges", update = FALSE)</span></span>
<span><span class="co"># BiocManager::install("plyranges", update = FALSE)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="use-excluderanges">Use excluderanges<a class="anchor" aria-label="anchor" href="#use-excluderanges"></a>
</h2>
<p>Get an overview of what’s available</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2022-10-31</span></span>
<span><span class="va">query_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">ah</span>, <span class="va">preparerclass</span> <span class="op">==</span> <span class="st">"excluderanges"</span><span class="op">)</span></span>
<span><span class="co"># You can search for multiple terms</span></span>
<span><span class="co"># query_data &lt;- query(ah, c("excluderanges", "Kundaje", "hg38"))</span></span>
<span><span class="va">query_data</span></span>
<span><span class="co">#&gt; AnnotationHub with 82 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-10-31</span></span>
<span><span class="co">#&gt; # $dataprovider: UCSC, GitHub, ENCODE, UCSChub, excluderanges, Stanford.edu,...</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Danio rerio...</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH107304"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                 </span></span>
<span><span class="co">#&gt;   AH107304 | T2T.excluderanges                     </span></span>
<span><span class="co">#&gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable</span></span>
<span><span class="co">#&gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 </span></span>
<span><span class="co">#&gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         </span></span>
<span><span class="co">#&gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        </span></span>
<span><span class="co">#&gt;   ...        ...                                   </span></span>
<span><span class="co">#&gt;   AH107381 | danRer10.UCSC.scaffold                </span></span>
<span><span class="co">#&gt;   AH107382 | dm6.UCSC.other                        </span></span>
<span><span class="co">#&gt;   AH107383 | dm3.UCSC.contig                       </span></span>
<span><span class="co">#&gt;   AH107384 | dm3.UCSC.scaffold                     </span></span>
<span><span class="co">#&gt;   AH107385 | TAIR10.UCSC.araTha1.gap</span></span></code></pre></div>
<p><code>hg38.Kundaje.GRCh38_unified_Excludable</code> object
recommended by Anshul</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">GenomeInfoDb</a></span><span class="op">)</span>  <span class="co"># for keepStandardChromosomes()</span></span>
<span><span class="va">excludeGR.hg38.Kundaje.1</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107305"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co"># Always a good idea to sort GRanges and keep standard chromosomes</span></span>
<span><span class="va">excludeGR.hg38.Kundaje.1</span> <span class="op">&lt;-</span> <span class="va">excludeGR.hg38.Kundaje.1</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html" class="external-link">keepStandardChromosomes</a></span><span class="op">(</span>pruning.mode <span class="op">=</span> <span class="st">"tidy"</span><span class="op">)</span></span>
<span><span class="va">excludeGR.hg38.Kundaje.1</span></span>
<span><span class="co">#&gt; GRanges object with 910 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames            ranges strand</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;     [1]     chr1     628903-635104      *</span></span>
<span><span class="co">#&gt;     [2]     chr1   5850087-5850571      *</span></span>
<span><span class="co">#&gt;     [3]     chr1   8909610-8910014      *</span></span>
<span><span class="co">#&gt;     [4]     chr1   9574580-9574997      *</span></span>
<span><span class="co">#&gt;     [5]     chr1 32043823-32044203      *</span></span>
<span><span class="co">#&gt;     ...      ...               ...    ...</span></span>
<span><span class="co">#&gt;   [906]     chrY 11290797-11334278      *</span></span>
<span><span class="co">#&gt;   [907]     chrY 11493053-11592850      *</span></span>
<span><span class="co">#&gt;   [908]     chrY 11671014-11671046      *</span></span>
<span><span class="co">#&gt;   [909]     chrY 11721528-11749472      *</span></span>
<span><span class="co">#&gt;   [910]     chrY 56694632-56889743      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from hg38 genome</span></span></code></pre></div>
<p>Save the data in a BED file, if needed.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note that rtracklayer::import and rtracklayer::export perform unexplained</span></span>
<span><span class="co"># start coordinate conversion, likely related to 0- and 1-based coordinate</span></span>
<span><span class="co"># system. We recommend converting GRanges to a data frame and save tab-separated</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span>, </span>
<span>            file <span class="op">=</span> <span class="st">"hg38.Kundaje.GRCh38_unified_Excludable.bed"</span>,</span>
<span>            sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>We can load other excludable regions for the hg38 genome assembly and
compare them.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"excluderanges"</span>, <span class="st">"hg38"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">query_data</span></span>
<span><span class="co">#&gt; AnnotationHub with 17 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-10-31</span></span>
<span><span class="co">#&gt; # $dataprovider: UCSC, ENCODE, GitHub, UCSChub</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH107305"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                 </span></span>
<span><span class="co">#&gt;   AH107305 | hg38.Kundaje.GRCh38_unified_Excludable</span></span>
<span><span class="co">#&gt;   AH107306 | hg38.Bernstein.Mint_Excludable_GRCh38 </span></span>
<span><span class="co">#&gt;   AH107307 | hg38.Boyle.hg38-Excludable.v2         </span></span>
<span><span class="co">#&gt;   AH107308 | hg38.Kundaje.GRCh38.Excludable        </span></span>
<span><span class="co">#&gt;   AH107309 | hg38.Lareau.hg38.full.Excludable      </span></span>
<span><span class="co">#&gt;   ...        ...                                   </span></span>
<span><span class="co">#&gt;   AH107355 | hg38.UCSC.telomere                    </span></span>
<span><span class="co">#&gt;   AH107356 | hg38.UCSC.short_arm                   </span></span>
<span><span class="co">#&gt;   AH107357 | hg38.UCSC.heterochromatin             </span></span>
<span><span class="co">#&gt;   AH107358 | hg38.UCSC.contig                      </span></span>
<span><span class="co">#&gt;   AH107359 | hg38.UCSC.scaffold</span></span>
<span><span class="va">excludeGR.hg38.Bernstein</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107306"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Boyle</span>     <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107307"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Kundaje.2</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107308"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Lareau</span>    <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107309"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Reddy</span>     <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107310"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Wimberley</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107311"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Wold</span>      <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107312"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">excludeGR.hg38.Yeo</span>       <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107313"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span></code></pre></div>
<div class="section level3">
<h3 id="compare-the-number-of-excludable-regions">Compare the number of excludable regions<a class="anchor" aria-label="anchor" href="#compare-the-number-of-excludable-regions"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Boyle</span><span class="op">)</span>,</span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Lareau</span><span class="op">)</span>,</span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wimberley</span><span class="op">)</span>,</span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                          Source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, </span>
<span>                                     <span class="st">"Boyle.hg38-Excludable.v2"</span>,</span>
<span>                                     <span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, </span>
<span>                                     <span class="st">"Kundaje.GRCh38.Excludable"</span>, </span>
<span>                                     <span class="st">"Lareau.hg38.full.Excludable"</span>,</span>
<span>                                     <span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, </span>
<span>                                     <span class="st">"Wimberley.peakPass60Perc_sorted"</span>,</span>
<span>                                     <span class="st">"Wold.hg38mitoExcludable"</span>, </span>
<span>                                     <span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover.bed"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Order Source by the number of regions</span></span>
<span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span>, levels <span class="op">=</span> <span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Count</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Source</span>, y <span class="op">=</span> <span class="va">Count</span>, fill <span class="op">=</span> <span class="va">Source</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/README-excluderanges_hg38_count-1.png" width="100%"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ggsave("../man/figures/excluderanges_hg38_count.png", width = 5.5, height = 2)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare-the-width-of-excludable-regions">Compare the width of excludable regions<a class="anchor" aria-label="anchor" href="#compare-the-width-of-excludable-regions"></a>
</h3>
<p>log2 scale because of heavy right tail distributions.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggridges/" class="external-link">ggridges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Boyle</span><span class="op">)</span>,</span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Lareau</span><span class="op">)</span>,</span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wimberley</span><span class="op">)</span>,</span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                          Source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Boyle.hg38-Excludable.v2"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Boyle</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Kundaje.GRCh38.Excludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Lareau.hg38.full.Excludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Lareau</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Wimberley.peakPass60Perc_sorted"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wimberley</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Wold.hg38mitoExcludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover.bed"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Order objects by decreasing width</span></span>
<span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span>, levels <span class="op">=</span> <span class="va">mtx_to_plot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>                               <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Source</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Width</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>                               <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Mean</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">Source</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">Width</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">Source</span>, fill <span class="op">=</span> <span class="va">Source</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://wilkelab.org/ggridges/reference/geom_density_ridges.html" class="external-link">geom_density_ridges</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/README-excluderanges_hg38_width-1.png" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ggsave("../man/figures/excluderanges_hg38_width.png", width = 5.5, height = 2)</span></span></code></pre></div>
<p>We can investigate the total width of each set of excludable
ranges.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>TotalWidth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Boyle</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Lareau</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wimberley</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>                          Source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, </span>
<span>                                     <span class="st">"Boyle.hg38-Excludable.v2"</span>,</span>
<span>                                     <span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, </span>
<span>                                     <span class="st">"Kundaje.GRCh38.Excludable"</span>, </span>
<span>                                     <span class="st">"Lareau.hg38.full.Excludable"</span>,</span>
<span>                                     <span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, </span>
<span>                                     <span class="st">"Wimberley.peakPass60Perc_sorted"</span>,</span>
<span>                                     <span class="st">"Wold.hg38mitoExcludable"</span>, </span>
<span>                                     <span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover.bed"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Order objects by decreasing width</span></span>
<span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span>, levels <span class="op">=</span> <span class="va">mtx_to_plot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>                               <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Source</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">TotalWidth</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">Source</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">TotalWidth</span>, y <span class="op">=</span> <span class="va">Source</span>, fill <span class="op">=</span> <span class="va">Source</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html" class="external-link">scale_y_discrete</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">abbreviate</span>, limits<span class="op">=</span><span class="va">rev</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"log10 total width"</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/README-excluderanges_hg38_sumwidth-1.png" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ggsave("../man/figures/excluderanges_hg38_sumwidth.png", width = 6.5, height = 2)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare-overlaps-among-sets">Compare overlaps among sets<a class="anchor" aria-label="anchor" href="#compare-overlaps-among-sets"></a>
</h3>
<p>We can compare <a href="https://en.wikipedia.org/wiki/Overlap_coefficient" class="external-link">overlap
coefficients</a> between those sets of excludable regions.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span></span>
<span><span class="co"># Overlap coefficient calculations</span></span>
<span><span class="va">overlap_coefficient</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr_a</span>, <span class="va">gr_b</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">intersects</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">gr_a</span>, <span class="va">gr_b</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">intersection_width</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">intersects</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">min_width</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gr_a</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">gr_b</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="va">intersection_width</span>, <span class="va">min_width</span>, </span>
<span>            overlap_coefficient <span class="op">=</span> <span class="va">intersection_width</span><span class="op">/</span><span class="va">min_width</span>,</span>
<span>             n_intersections <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">intersects</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># List and names of all excludable regions</span></span>
<span><span class="va">all_excludeGR_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span>, </span>
<span>                            <span class="va">excludeGR.hg38.Boyle</span>,</span>
<span>                            <span class="va">excludeGR.hg38.Kundaje.1</span>, </span>
<span>                            <span class="va">excludeGR.hg38.Kundaje.2</span>,</span>
<span>                            <span class="va">excludeGR.hg38.Lareau</span>,</span>
<span>                            <span class="va">excludeGR.hg38.Reddy</span>,</span>
<span>                            <span class="va">excludeGR.hg38.Wimberley</span>,</span>
<span>                            <span class="va">excludeGR.hg38.Wold</span>,</span>
<span>                            <span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span></span>
<span><span class="va">all_excludeGR_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, </span>
<span>                         <span class="st">"Boyle.hg38-Excludable.v2"</span>,</span>
<span>                         <span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, </span>
<span>                         <span class="st">"Kundaje.GRCh38.Excludable"</span>, </span>
<span>                         <span class="st">"Lareau.hg38.full.Excludable"</span>,</span>
<span>                         <span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, </span>
<span>                         <span class="st">"Wimberley.peakPass60Perc_sorted"</span>,</span>
<span>                         <span class="st">"Wold.hg38mitoExcludable"</span>, </span>
<span>                         <span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover.bed"</span><span class="op">)</span></span>
<span><span class="co"># Correlation matrix, empty</span></span>
<span><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fl">0</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Fill it in</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># If diagonal, set to zero</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">i</span> <span class="op">==</span> <span class="va">j</span><span class="op">)</span> <span class="va">mtx_to_plot</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span>    <span class="co"># Process only one half, the other is symmetric</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">i</span> <span class="op">&gt;</span> <span class="va">j</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">mtx_to_plot</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mtx_to_plot</span><span class="op">[</span><span class="va">j</span>, <span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">overlap_coefficient</span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="va">all_excludeGR_list</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="st">"overlap_coefficient"</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># Trim row/colnames</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_trunc.html" class="external-link">str_trunc</a></span><span class="op">(</span><span class="va">all_excludeGR_name</span>, width <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> </span>
<span><span class="co"># Save the plot</span></span>
<span><span class="co"># png("../man/figures/excluderanges_hg38_jaccard.png", width = 1000, height = 900, res = 200)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html" class="external-link">data.matrix</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span>, clustering_method <span class="op">=</span> <span class="st">"ward.D"</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/README-excluderanges_hg38_overlap_coefficient-1.png" width="100%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># dev.off()</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="metadata-analysis">Metadata analysis<a class="anchor" aria-label="anchor" href="#metadata-analysis"></a>
</h3>
<p>Note that some excludable ranges objects contain six columns,
implying there may be some interesting metadata. Let’s explore one.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 396 rows and 2 columns</span></span>
<span><span class="co">#&gt;                       name     score</span></span>
<span><span class="co">#&gt;                &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; 1   High_Mappability_isl..      1000</span></span>
<span><span class="co">#&gt; 2         Satellite_repeat      1000</span></span>
<span><span class="co">#&gt; 3                 BSR/Beta      1000</span></span>
<span><span class="co">#&gt; 4   Low_mappability_island      1000</span></span>
<span><span class="co">#&gt; 5                 (CATTC)n      1000</span></span>
<span><span class="co">#&gt; ...                    ...       ...</span></span>
<span><span class="co">#&gt; 392       Satellite_repeat      1000</span></span>
<span><span class="co">#&gt; 393               (CATTC)n      1000</span></span>
<span><span class="co">#&gt; 394               (CATTC)n      1000</span></span>
<span><span class="co">#&gt; 395                   TAR1      1000</span></span>
<span><span class="co">#&gt; 396                   chrM      1000</span></span>
<span><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span><span class="op">[[</span><span class="st">"name"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Type"</span>, <span class="st">"Number"</span><span class="op">)</span></span>
<span><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="va">mtx_to_plot</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Number</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Type</span>, </span>
<span>                           levels <span class="op">=</span> <span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Type</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Number</span>, y <span class="op">=</span> <span class="va">Type</span>, fill <span class="op">=</span> <span class="va">Type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<p><img src="../reference/figures/README-excluderanges_hg38_Reddy_metadata-1.png" width="100%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># ggsave("../man/figures/excluderanges_hg38_Reddy_metadata.png", width = 5, height = 2.5)</span></span></code></pre></div>
<p>One may decide to combine the excludable ranges from all labs,
although from previous results we may decide to follow Anshul’s <a href="https://twitter.com/anshulkundaje/status/1263546023151992832?s=20" class="external-link">advice</a>
about the <a href="https://www.encodeproject.org/files/ENCFF356LFX/" class="external-link">ENCFF356LFX
exclusion list regions</a> and use the
<code>excludeGR.hg38.Kundaje.1</code> object.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">excludeGR.hg38.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span>, </span>
<span>                               <span class="va">excludeGR.hg38.Boyle</span>,</span>
<span>                               <span class="va">excludeGR.hg38.Kundaje.1</span>, </span>
<span>                               <span class="va">excludeGR.hg38.Kundaje.2</span>, </span>
<span>                               <span class="va">excludeGR.hg38.Lareau</span>,</span>
<span>                               <span class="va">excludeGR.hg38.Reddy</span>, </span>
<span>                               <span class="va">excludeGR.hg38.Wimberley</span>,</span>
<span>                               <span class="va">excludeGR.hg38.Wold</span>, </span>
<span>                               <span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Sort and Keep only standard chromosomes</span></span>
<span><span class="va">excludeGR.hg38.all</span> <span class="op">&lt;-</span> <span class="va">excludeGR.hg38.all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">sort</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html" class="external-link">keepStandardChromosomes</a></span><span class="op">(</span>pruning.mode <span class="op">=</span> <span class="st">"tidy"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.all</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 15998</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/start.html" class="external-link">width</a></span><span class="op">(</span><span class="va">excludeGR.hg38.all</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span><span class="co">#&gt;        5     1703     2294    17713     3001 30739401</span></span></code></pre></div>
<!--# BEDbase data download

[BEDbase.org](http://bedbase.org/) is a repository for storing and analyzing
BED files and BED sets, developed by [Sheffield lab](https://databio.org/). 
It provides API for data access and retrieval. 

Using BEDbase ID (e.g., `1a561729234c2844303a051b16f66656` for the 
`hg38.Kundaje.GRCh38_unified_Excludable` excluderanges object), we can construct
a URL showing the splash screen with detailed information of the corresponding
object, http://bedbase.org/#/bedsplash/1a561729234c2844303a051b16f66656.

We can also get the BED file using the [BEDbase API](http://bedbase.org/docs).


```bash
wget http://bedbase.org/api/bed/1a561729234c2844303a051b16f66656/file/bed \
  -O hg38.Kundaje.GRCh38_unified_Excludable.bed.gz 
```

See also the R code to download
[T2T.excluderanges](https://gist.github.com/mdozmorov/7f4393f90932fb6bd911c43c20425ca0) 
and [mm39.excluderanges](https://gist.github.com/mdozmorov/33a1fa3234b2942dae238e1fcb39c996).
-->
</div>
</div>
<div class="section level2">
<h2 id="mitochondrial-dna-sequences-numts">Mitochondrial DNA sequences, NUMTs<a class="anchor" aria-label="anchor" href="#mitochondrial-dna-sequences-numts"></a>
</h2>
<p>Mitochondrial DNA sequences (mtDNA, 100-600K mitochondria per human
cell) transferred to the nucleus give rise to the so-called
mitochondrial DNA sequences in the nuclear genome (NUMTs). In the
settings of DNA/chromatin sequencing (e.g., ATAC-seq), we may observe up
to 80% of mitochondrial sequencing reads that may pile up in the NUMT
sequences. Similar to exclusion sets, genomic regions highly homologous
to mtDNA can be masked to improve biological signal.</p>
<p>The reference human nuclear mitochondrial sequences have been
available in the UCSC genome browser for hg19 and mm8 human/mouse genome
assemblies. We collected NUMT sets for hg38, T2T-CHM13, mm10, generated
by Caleb Lareau in the <a href="https://github.com/caleblareau/mitoblacklist" class="external-link">mitoblacklist</a>
GitHub repository. These NUMT sets can be combined with exclusion
sets.</p>
<p>Example of the <code>hg38.Lareau.hg38_peaks</code> object</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query_data</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"mito"</span>, <span class="va">query_data</span><span class="op">$</span><span class="va">description</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-10-31</span></span>
<span><span class="co">#&gt; # names(): AH107343</span></span>
<span><span class="co">#&gt; # $dataprovider: GitHub</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2022-10-19</span></span>
<span><span class="co">#&gt; # $title: hg38.Lareau.hg38_peaks</span></span>
<span><span class="co">#&gt; # $description: Regions of high homology to mtDNA (NUMT regions) defined by ...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: hg38</span></span>
<span><span class="co">#&gt; # $sourcetype: RDS</span></span>
<span><span class="co">#&gt; # $sourceurl: https://drive.google.com/drive/folders/1sF9m8Y3eZouTZ3IEEywjs2...</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("FunctionalAnnotation", "GenomicSequence",</span></span>
<span><span class="co">#&gt; #   "hg38;Lareau;NUMTs") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH107343"]]'</span></span>
<span><span class="va">hg38.Lareau.hg38_peaks</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107343"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="centromeres-telomeres-etc-">Centromeres, telomeres, etc.<a class="anchor" aria-label="anchor" href="#centromeres-telomeres-etc-"></a>
</h2>
<p>Besides the ENCODE-produced excludable regions, we may want to
exclude centromeres, telomeres, and other gap locations. The “Gap
Locations” track for Homo Sapiens is available for the GRcH37/hg19
genome assembly as a <a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="external-link">UCSC
‘gap’ table</a>. It can be retrieved from <em><a href="https://bioconductor.org/packages/3.16/AnnotationHub" class="external-link">AnnotationHub</a></em>,
but lacks the metadata columns needed to decide the type of gaps.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Search for the gap track</span></span>
<span><span class="co"># ahData &lt;- query(ah, c("gap", "Homo sapiens", "hg19"))</span></span>
<span><span class="co"># ahData[ahData$title == "Gap"]</span></span>
<span><span class="va">gaps</span> <span class="op">&lt;-</span> <span class="va">ahData</span><span class="op">[[</span><span class="st">"AH6444"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p>The <a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="external-link">UCSC
‘gap’ table</a> provides better granularity about the types of gaps
available. E.g., for human, hg19, we have the following types and the
number of gaps.</p>
<p><img src="../reference/figures/README-gapcounts-1.png" width="70%" height="70%"></p>
<p>Those objects are provided as individual GRanges.</p>
<p>Naming convention:
<code>&lt;genome assembly&gt;.UCSC.&lt;gap type&gt;</code>, e.g.,
<code>hg38.UCSC.gap_centromere</code>. We can similarly load any gap
type object.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"excluderanges"</span>, <span class="st">"UCSC"</span>, <span class="st">"Homo Sapiens"</span>, <span class="st">"hg38"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">query_data</span></span>
<span><span class="co">#&gt; AnnotationHub with 7 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-10-31</span></span>
<span><span class="co">#&gt; # $dataprovider: UCSC, UCSChub</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH107353"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                    </span></span>
<span><span class="co">#&gt;   AH107353 | T2T.UCSC.hgUnique.hg38   </span></span>
<span><span class="co">#&gt;   AH107354 | hg38.UCSC.centromere     </span></span>
<span><span class="co">#&gt;   AH107355 | hg38.UCSC.telomere       </span></span>
<span><span class="co">#&gt;   AH107356 | hg38.UCSC.short_arm      </span></span>
<span><span class="co">#&gt;   AH107357 | hg38.UCSC.heterochromatin</span></span>
<span><span class="co">#&gt;   AH107358 | hg38.UCSC.contig         </span></span>
<span><span class="co">#&gt;   AH107359 | hg38.UCSC.scaffold</span></span>
<span></span>
<span><span class="va">gapsGR_hg38_centromere</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH107354"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">gapsGR_hg38_centromere</span></span>
<span><span class="co">#&gt; GRanges object with 109 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames              ranges strand |       bin        name</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;     [1]     chr1 122026459-122224535      * |       189  GJ211836.1</span></span>
<span><span class="co">#&gt;     [2]     chr1 122224635-122503147      * |       189  GJ211837.1</span></span>
<span><span class="co">#&gt;     [3]     chr1 122503247-124785432      * |        23  GJ212202.1</span></span>
<span><span class="co">#&gt;     [4]     chr1 124785532-124849129      * |      1537  GJ211855.1</span></span>
<span><span class="co">#&gt;     [5]     chr1 124849229-124932724      * |       192  GJ211857.1</span></span>
<span><span class="co">#&gt;     ...      ...                 ...    ... .       ...         ...</span></span>
<span><span class="co">#&gt;   [105]    chr22   14419994-14420334      * |       695  GJ212209.1</span></span>
<span><span class="co">#&gt;   [106]    chr22   14420434-14421632      * |       695  GJ212186.2</span></span>
<span><span class="co">#&gt;   [107]    chr22   14421732-15054318      * |        10  GJ212191.2</span></span>
<span><span class="co">#&gt;   [108]     chrX   58605579-62412542      * |         1  GJ212192.1</span></span>
<span><span class="co">#&gt;   [109]     chrY   10316944-10544039      * |        10  GJ212193.1</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from hg38 genome</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cutrun-excludable-sets">CUT&amp;RUN excludable sets<a class="anchor" aria-label="anchor" href="#cutrun-excludable-sets"></a>
</h2>
<p>Nordin et al. 2022 <span class="citation">(Nordin et al. 2022)</span>
generated excludable regions for the CUT&amp;RUN technology. They are
available as <a href="https://www.biorxiv.org/content/10.1101/2022.11.11.516118v1.supplementary-material" class="external-link">Supplementary
Material</a>. We uniformly processed them and made available on Google
Drive</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># hg38 CUT&amp;RUN exclusion set, BED</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://drive.google.com/uc?export=download&amp;id=1rKIu7kdiEySTi-cq3nYxXJP4VQX1IPcS"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="st">"hg38.Nordin.CandRblacklist_hg38.bed"</span><span class="op">)</span></span>
<span><span class="co"># hg38 CUT&amp;RUN exclusion set, RDS</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://drive.google.com/uc?export=download&amp;id=1JuB1h-QQUw1mddBavI7CIuH7R-lwwczU"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="st">"hg38.Nordin.CandRblacklist_hg38.rds"</span><span class="op">)</span></span>
<span><span class="co"># And then load the GRanges object</span></span>
<span><span class="va">mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"hg38.Nordin.CandRblacklist_hg38.rds"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># mm10 CUT&amp;RUN exclusion set, BED</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://drive.google.com/uc?export=download&amp;id=1CRAojdphMbAzd3MnW_UmO1WtsDrHsrU1"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="st">"mm10.Nordin.CandRblacklist_mm10.bed"</span><span class="op">)</span></span>
<span><span class="co"># mm10 CUT&amp;RUN exclusion set, RDS</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://drive.google.com/uc?export=download&amp;id=1orPXLWUZ4-C4n_Jt2gH-WERLpY9Kn0t_"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="st">"mm10.Nordin.CandRblacklist_mm10.rds"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="summary-table">Summary table<a class="anchor" aria-label="anchor" href="#summary-table"></a>
</h2>
<p><a href="../man/figures/Table_S1.csv">Full summary table</a>.</p>
<table class="table">
<colgroup>
<col width="18%">
<col width="6%">
<col width="69%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">Name</th>
<th align="left">Ahub.IDs.BioC.3.16.and.above</th>
<th align="left">Description</th>
<th align="right">Filtered.Region.count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">T2T.excluderanges</td>
<td align="left">AH107304</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">3565</td>
</tr>
<tr class="even">
<td align="left">hg38.Kundaje.GRCh38_unified_Excludable</td>
<td align="left">AH107305</td>
<td align="left">Defined as a combination of hg38.Lareau.hg38_peaks,
hg38.Boyle.hg38-Excludable.v2, and hg38.Wimberley.peakPass60Perc_sorted,
followed by manual curation. Supersedes
hg38.Kundaje.GRCh38.Excludable.</td>
<td align="right">910</td>
</tr>
<tr class="odd">
<td align="left">hg38.Bernstein.Mint_Excludable_GRCh38</td>
<td align="left">AH107306</td>
<td align="left">Defined from Mint-ChIP (low input, multiplexed
ChIP-seq) data</td>
<td align="right">12052</td>
</tr>
<tr class="even">
<td align="left">hg38.Boyle.hg38-Excludable.v2</td>
<td align="left">AH107307</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">636</td>
</tr>
<tr class="odd">
<td align="left">hg38.Kundaje.GRCh38.Excludable</td>
<td align="left">AH107308</td>
<td align="left">Defined by Anshul Kundaje as a part of ENCODE and
modENCODE consortia</td>
<td align="right">38</td>
</tr>
<tr class="even">
<td align="left">hg38.Lareau.hg38.full.Excludable</td>
<td align="left">AH107309</td>
<td align="left">ENCODE excludable regions combined with regions of high
homology to mtDNA (NUMT regions)</td>
<td align="right">820</td>
</tr>
<tr class="odd">
<td align="left">hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38</td>
<td align="left">AH107310</td>
<td align="left">Defined by the ENCODE consortium, includes satellite
repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha,
BSR/Beta), centromeric repeats, chrM, High/Low mappability islands. Has
extra chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">396</td>
</tr>
<tr class="even">
<td align="left">hg38.Wimberley.peakPass60Perc_sorted</td>
<td align="left">AH107311</td>
<td align="left">Defined by the ewimberley/peakPass software</td>
<td align="right">5078</td>
</tr>
<tr class="odd">
<td align="left">hg38.Wold.hg38mitoExcludable</td>
<td align="left">AH107312</td>
<td align="left">Definition method unknown</td>
<td align="right">299</td>
</tr>
<tr class="even">
<td align="left">hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed</td>
<td align="left">AH107313</td>
<td align="left">Defined from eCLIP data</td>
<td align="right">56</td>
</tr>
<tr class="odd">
<td align="left">hg38.Nordin.CandRblacklist_hg38</td>
<td align="left">NA</td>
<td align="left">Defined from CUT&amp;RUN negative controls as 0.1% top
significant SEACR peaks in over 30% of samples</td>
<td align="right">885</td>
</tr>
<tr class="even">
<td align="left">hg19.Boyle.hg19-Excludable.v2</td>
<td align="left">AH107314</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">834</td>
</tr>
<tr class="odd">
<td align="left">hg19.Bernstein.Mint_Excludable_hg19</td>
<td align="left">AH107315</td>
<td align="left">Defined from Mint-ChIP (low input, multiplexed
ChIP-seq) data</td>
<td align="right">9035</td>
</tr>
<tr class="even">
<td align="left">hg19.Birney.wgEncodeDacMapabilityConsensusExcludable</td>
<td align="left">AH107316</td>
<td align="left">Defined by the ENCODE consortium, includes satellite
repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha,
BSR/Beta), centromeric repeats, chrM, High/Low mappability islands</td>
<td align="right">411</td>
</tr>
<tr class="odd">
<td align="left">hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable</td>
<td align="left">AH107317</td>
<td align="left">Defined by the ENCODE consortium, includes satellite
repeats (CATTC, GAATG, GAGTG, ACRO1), RepeatMasker repeats (ALR/Alpha,
BSR/Beta), human satellite repeat HSATII, chrM, ribosomal subunit
consensus sequences LSU-rRNA_Hsa, SSU-rRNA_Hsa. Has extra chromosomes,
use keepStandardChromosomes() filtering</td>
<td align="right">1566</td>
</tr>
<tr class="even">
<td align="left">hg19.Lareau.hg19.full.Excludable</td>
<td align="left">AH107318</td>
<td align="left">ENCODE excludable regions combined with regions of high
homology to mtDNA (NUMT regions)</td>
<td align="right">902</td>
</tr>
<tr class="odd">
<td align="left">hg19.Wold.hg19mitoExcludable</td>
<td align="left">AH107319</td>
<td align="left">Definition method unknown</td>
<td align="right">295</td>
</tr>
<tr class="even">
<td align="left">hg19.Yeo.eCLIP_Excludableregions.hg19</td>
<td align="left">AH107320</td>
<td align="left">Defined from eCLIP data, includes skyscraper, rRNA
pseudogene, unreliably mapped satellite repeat, and low complexity
skyscraper peak regions</td>
<td align="right">57</td>
</tr>
<tr class="odd">
<td align="left">mm39.excluderanges</td>
<td align="left">AH107321</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">3147</td>
</tr>
<tr class="even">
<td align="left">mm10.Boyle.mm10-Excludable.v2</td>
<td align="left">AH107322</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">3435</td>
</tr>
<tr class="odd">
<td align="left">mm10.Hardison.Excludable.full</td>
<td align="left">AH107323</td>
<td align="left">Definition method unknown</td>
<td align="right">7865</td>
</tr>
<tr class="even">
<td align="left">mm10.Hardison.psuExcludable.mm10</td>
<td align="left">AH107324</td>
<td align="left">Definition method unknown</td>
<td align="right">5552</td>
</tr>
<tr class="odd">
<td align="left">mm10.Kundaje.anshul.Excludable.mm10</td>
<td align="left">AH107325</td>
<td align="left">Defined by Anshul Kundaje as a part of ENCODE and
modENCODE consortia</td>
<td align="right">3010</td>
</tr>
<tr class="even">
<td align="left">mm10.Kundaje.mm10.Excludable</td>
<td align="left">AH107326</td>
<td align="left">Defined by Anshul Kundaje as a part of ENCODE and
modENCODE consortia</td>
<td align="right">164</td>
</tr>
<tr class="odd">
<td align="left">mm10.Lareau.mm10.full.Excludable</td>
<td align="left">AH107327</td>
<td align="left">ENCODE excludable regions combined with regions of high
homology to mtDNA (NUMT regions)</td>
<td align="right">523</td>
</tr>
<tr class="even">
<td align="left">mm10.Wold.mm10mitoExcludable</td>
<td align="left">AH107328</td>
<td align="left">Definition method unknown</td>
<td align="right">123</td>
</tr>
<tr class="odd">
<td align="left">mm10.Nordin.CandRblacklist_mm10</td>
<td align="left">NA</td>
<td align="left">Defined from CUT&amp;RUN negative controls as 0.1% top
significant SEACR peaks in over 30% of samples</td>
<td align="right">559</td>
</tr>
<tr class="even">
<td align="left">mm9.Lareau.mm9.full.Excludable</td>
<td align="left">AH107329</td>
<td align="left">ENCODE excludable regions combined with regions of high
homology to mtDNA (NUMT regions)</td>
<td align="right">3415</td>
</tr>
<tr class="odd">
<td align="left">mm9.Wold.mm9mitoExcludable</td>
<td align="left">AH107330</td>
<td align="left">Definition method unknown</td>
<td align="right">123</td>
</tr>
<tr class="even">
<td align="left">ce11.Boyle.ce11-Excludable.v2</td>
<td align="left">AH107331</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">97</td>
</tr>
<tr class="odd">
<td align="left">ce10.Boyle.ce10-Excludable.v2</td>
<td align="left">AH107332</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">100</td>
</tr>
<tr class="even">
<td align="left">ce10.Kundaje.ce10-Excludable</td>
<td align="left">AH107333</td>
<td align="left">Defined by Anshul Kundaje, superseded by
ce10.Boyle.ce10-Excludable.v2</td>
<td align="right">122</td>
</tr>
<tr class="odd">
<td align="left">danRer10.Domingues.Excludableed</td>
<td align="left">AH107334</td>
<td align="left">Defined manually using total RNA-seq.</td>
<td align="right">57</td>
</tr>
<tr class="even">
<td align="left">danRer10.Yang.Supplemental_Table_19.ChIP-seq_black_list_in_the_zebrafish_genome</td>
<td align="left">AH107335</td>
<td align="left">Defined via MACS2 peak calling using ChIP-seq (PMID:
33239788)</td>
<td align="right">853</td>
</tr>
<tr class="odd">
<td align="left">dm6.Boyle.dm6-Excludable.v2</td>
<td align="left">AH107336</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">182</td>
</tr>
<tr class="even">
<td align="left">dm3.Boyle.dm3-Excludable.v2</td>
<td align="left">AH107337</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions</td>
<td align="right">248</td>
</tr>
<tr class="odd">
<td align="left">dm3.Kundaje.dm3-Excludable</td>
<td align="left">AH107338</td>
<td align="left">Defined by Anshul Kundaje. Contains heterochromatin
chromosomes chr2LHet. Superseded by dm3.Boyle.dm3-Excludable.v2</td>
<td align="right">306</td>
</tr>
<tr class="even">
<td align="left">TAIR10.Wimberley.predicted_excluded_list_sorted_0.6</td>
<td align="left">AH107339</td>
<td align="left">Defined by the ewimberley/peakPass software</td>
<td align="right">887</td>
</tr>
<tr class="odd">
<td align="left">TAIR10.Klasfeld.arabidopsis_Excludable_20inputs</td>
<td align="left">AH107340</td>
<td align="left">Defined by the Boyle-Lab/Blacklist software, High
Signal and Low Mappability regions (DOI: 10.1101/2022.02.27.482177)</td>
<td align="right">83</td>
</tr>
<tr class="even">
<td align="left">TAIR10.Klasfeld.arabidopsis_greenscreen_20inputs</td>
<td align="left">AH107341</td>
<td align="left">Defined by the green screen pipeline (DOI:
10.1101/2022.02.27.482177)</td>
<td align="right">36</td>
</tr>
<tr class="odd">
<td align="left">T2T.Lareau.chm13v2.0_peaks</td>
<td align="left">AH107342</td>
<td align="left">Regions of high homology to mtDNA (NUMT regions)
defined by caleblareau/mitoblacklist</td>
<td align="right">817</td>
</tr>
<tr class="even">
<td align="left">hg38.Lareau.hg38_peaks</td>
<td align="left">AH107343</td>
<td align="left">Regions of high homology to mtDNA (NUMT regions)
defined by caleblareau/mitoblacklist</td>
<td align="right">784</td>
</tr>
<tr class="odd">
<td align="left">hg19.Lareau.hg19_peaks</td>
<td align="left">AH107344</td>
<td align="left">Regions of high homology to mtDNA (NUMT regions)
defined by caleblareau/mitoblacklist</td>
<td align="right">779</td>
</tr>
<tr class="even">
<td align="left">mm10.Lareau.mm10_peaks</td>
<td align="left">AH107345</td>
<td align="left">Regions of high homology to mtDNA (NUMT regions)
defined by caleblareau/mitoblacklist</td>
<td align="right">387</td>
</tr>
<tr class="odd">
<td align="left">mm9.Lareau.mm9_peaks</td>
<td align="left">AH107346</td>
<td align="left">Regions of high homology to mtDNA (NUMT regions)
defined by caleblareau/mitoblacklist</td>
<td align="right">395</td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.numtS</td>
<td align="left">AH107347</td>
<td align="left">Human NumtS mitochondrial sequence</td>
<td align="right">766</td>
</tr>
<tr class="odd">
<td align="left">mm9.UCSC.numtS</td>
<td align="left">AH107348</td>
<td align="left">Mouse NumtS mitochondrial sequence</td>
<td align="right">172</td>
</tr>
<tr class="even">
<td align="left">T2T.CHM13.chm13.draft_v2.0.cen_mask</td>
<td align="left">AH107349</td>
<td align="left">Centromeric satellite masking bed file (v2.0)</td>
<td align="right">23</td>
</tr>
<tr class="odd">
<td align="left">T2T.CHM13.chm13.draft_v1.1.telomere</td>
<td align="left">AH107350</td>
<td align="left">Telomere identified by the VGP pipeline (v1.1)</td>
<td align="right">48</td>
</tr>
<tr class="even">
<td align="left">T2T.UCSC.censat</td>
<td align="left">AH107351</td>
<td align="left">T2T peri/centromeric satellite annotation (v2.0,
20220329, CHM13 v2.0)</td>
<td align="right">2523</td>
</tr>
<tr class="odd">
<td align="left">T2T.UCSC.gap</td>
<td align="left">AH107352</td>
<td align="left">Locations of assembly gaps, as determine by strings of
‘N’ characters (v1.0)</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">T2T.UCSC.hgUnique.hg38</td>
<td align="left">AH107353</td>
<td align="left">Regions unique to the T2T-CHM13 v2.0 assembly compared
to the GRCh38/hg38 and GRCh37/hg19 reference assemblies</td>
<td align="right">615</td>
</tr>
<tr class="odd">
<td align="left">hg38.UCSC.centromere</td>
<td align="left">AH107354</td>
<td align="left">Gaps from centromeres</td>
<td align="right">109</td>
</tr>
<tr class="even">
<td align="left">hg38.UCSC.telomere</td>
<td align="left">AH107355</td>
<td align="left">Gaps from telomeres</td>
<td align="right">48</td>
</tr>
<tr class="odd">
<td align="left">hg38.UCSC.short_arm</td>
<td align="left">AH107356</td>
<td align="left">Gaps on the short arm of the chromosome</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">hg38.UCSC.heterochromatin</td>
<td align="left">AH107357</td>
<td align="left">Gaps from large blocks of heterochromatin</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">hg38.UCSC.contig</td>
<td align="left">AH107358</td>
<td align="left">Gaps between contigs in scaffolds</td>
<td align="right">285</td>
</tr>
<tr class="even">
<td align="left">hg38.UCSC.scaffold</td>
<td align="left">AH107359</td>
<td align="left">Gaps between scaffolds in chromosome assemblies. Has
extra chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">254</td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.centromere</td>
<td align="left">AH107360</td>
<td align="left">Gaps from centromeres</td>
<td align="right">24</td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.telomere</td>
<td align="left">AH107361</td>
<td align="left">Gaps from telomeres</td>
<td align="right">46</td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.short_arm</td>
<td align="left">AH107362</td>
<td align="left">Gaps on the short arm of the chromosome</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.heterochromatin</td>
<td align="left">AH107363</td>
<td align="left">Gaps from large blocks of heterochromatin</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.clone</td>
<td align="left">AH107364</td>
<td align="left">Gaps between clones in the same map contig. Has extra
chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">107</td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.contig</td>
<td align="left">AH107365</td>
<td align="left">Gaps between contigs in scaffolds</td>
<td align="right">163</td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.scaffold</td>
<td align="left">AH107366</td>
<td align="left">Gaps between scaffolds in chromosome assemblies. Only
non-autosomal chromosomes</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">mm39.UCSC.centromere</td>
<td align="left">AH107367</td>
<td align="left">Gaps from centromeres</td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">mm39.UCSC.telomere</td>
<td align="left">AH107368</td>
<td align="left">Gaps from telomeres</td>
<td align="right">42</td>
</tr>
<tr class="even">
<td align="left">mm39.UCSC.short_arm</td>
<td align="left">AH107369</td>
<td align="left">Gaps on the short arm of the chromosome</td>
<td align="right">21</td>
</tr>
<tr class="odd">
<td align="left">mm39.UCSC.contig</td>
<td align="left">AH107370</td>
<td align="left">Gaps between contigs in scaffolds</td>
<td align="right">60</td>
</tr>
<tr class="even">
<td align="left">mm39.UCSC.scaffold</td>
<td align="left">AH107371</td>
<td align="left">Gaps between scaffolds in chromosome assemblies</td>
<td align="right">115</td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.centromere</td>
<td align="left">AH107372</td>
<td align="left">Gaps from centromeres</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.telomere</td>
<td align="left">AH107373</td>
<td align="left">Gaps from telomeres</td>
<td align="right">42</td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.short_arm</td>
<td align="left">AH107374</td>
<td align="left">Gaps on the short arm of the chromosome</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.clone</td>
<td align="left">AH107375</td>
<td align="left">Gaps between clones in the same map contig. Has extra
chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.contig</td>
<td align="left">AH107376</td>
<td align="left">Gaps between contigs in scaffolds</td>
<td align="right">104</td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.scaffold</td>
<td align="left">AH107377</td>
<td align="left">Gaps between scaffolds in chromosome assemblies</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.other</td>
<td align="left">AH107378</td>
<td align="left">Sequence of Ns in the assembly that were not marked as
gaps in the AGP (A Golden Path) assembly definition file. Has extra
chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">383</td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.fragment</td>
<td align="left">AH107379</td>
<td align="left">A single gap of 31 bases in chrX_GL456233_random</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">mm9.UCSC.centromere</td>
<td align="left">AH107380</td>
<td align="left">Gaps from centromeres</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">mm9.UCSC.fragment</td>
<td align="left">AH107381</td>
<td align="left">Gaps between the contigs of a draft clone. (In this
context, a contig is a set of overlapping sequence reads). Has extra
chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">436</td>
</tr>
<tr class="odd">
<td align="left">mm9.UCSC.contig</td>
<td align="left">AH107382</td>
<td align="left">Gaps between contigs in scaffolds. Has extra
chromosomes, use keepStandardChromosomes() filtering</td>
<td align="right">105</td>
</tr>
<tr class="even">
<td align="left">danRer10.UCSC.contig</td>
<td align="left">AH107383</td>
<td align="left">Gaps between contigs in scaffolds</td>
<td align="right">2338</td>
</tr>
<tr class="odd">
<td align="left">danRer10.UCSC.scaffold</td>
<td align="left">AH107384</td>
<td align="left">Gaps between scaffolds in chromosome assemblies</td>
<td align="right">16496</td>
</tr>
<tr class="even">
<td align="left">dm6.UCSC.other</td>
<td align="left">AH107385</td>
<td align="left">Sequence of Ns in the assembly that were not marked as
gaps in the AGP (A Golden Path) assembly definition file</td>
<td align="right">268</td>
</tr>
<tr class="odd">
<td align="left">dm3.UCSC.contig</td>
<td align="left">NA</td>
<td align="left">Gaps between contigs in scaffolds</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">dm3.UCSC.scaffold</td>
<td align="left">NA</td>
<td align="left">Gaps between scaffolds in chromosome assemblies</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">TAIR10.UCSC.araTha1.gap</td>
<td align="left">NA</td>
<td align="left">Gaps in the May 2011 Arabidopsis thaliana genome
assembly</td>
<td align="right">357</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Below is the citation output from using
<code>citation('excluderanges')</code> in R. Please run this yourself to
check for any updates on how to cite <strong>excluderanges</strong>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"excluderanges"</span><span class="op">)</span>, bibtex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To cite package 'excluderanges' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Dozmorov MG, Davis E, Mu W, Lee S, Triche T, Phanstiel D, Love M</span></span>
<span><span class="co">#&gt;   (2025). _excluderanges_.</span></span>
<span><span class="co">#&gt;   https://github.com/dozmorovlab/excluderanges/excluderanges - R</span></span>
<span><span class="co">#&gt;   package version 0.99.10,</span></span>
<span><span class="co">#&gt;   &lt;https://github.com/dozmorovlab/excluderanges&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {excluderanges},</span></span>
<span><span class="co">#&gt;     author = {Mikhail G. Dozmorov and Eric Davis and Wancen Mu and Stuart Lee and Tim Triche and Douglas Phanstiel and Michael Love},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     url = {https://github.com/dozmorovlab/excluderanges},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/dozmorovlab/excluderanges/excluderanges - R package version 0.99.10},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the <code>excluderanges</code> project is released
with a <a href="https://bioconductor.github.io/bioc_coc_multilingual/" class="external-link">Contributor
Code of Conduct</a>. By contributing to this project, you agree to abide
by its terms.</p>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.16/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] rtracklayer_1.58.0   stringr_1.5.0        pheatmap_1.0.12     </span></span>
<span><span class="co">#&gt;  [4] dplyr_1.1.1          ggridges_0.5.4       ggplot2_3.4.2       </span></span>
<span><span class="co">#&gt;  [7] AnnotationHub_3.6.0  BiocFileCache_2.6.1  dbplyr_2.3.2        </span></span>
<span><span class="co">#&gt; [10] GenomicRanges_1.50.2 GenomeInfoDb_1.34.9  IRanges_2.32.0      </span></span>
<span><span class="co">#&gt; [13] S4Vectors_0.36.2     BiocGenerics_0.44.0  BiocStyle_2.26.0    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] matrixStats_0.63.0            bitops_1.0-7                 </span></span>
<span><span class="co">#&gt;  [3] fs_1.6.1                      bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [5] filelock_1.0.2                RColorBrewer_1.1-3           </span></span>
<span><span class="co">#&gt;  [7] httr_1.4.5                    rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt;  [9] tools_4.2.3                   bslib_0.4.2                  </span></span>
<span><span class="co">#&gt; [11] utf8_1.2.3                    R6_2.5.1                     </span></span>
<span><span class="co">#&gt; [13] DBI_1.1.3                     colorspace_2.1-0             </span></span>
<span><span class="co">#&gt; [15] withr_2.5.0                   tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt; [17] bit_4.0.5                     curl_5.0.0                   </span></span>
<span><span class="co">#&gt; [19] compiler_4.2.3                textshaping_0.3.6            </span></span>
<span><span class="co">#&gt; [21] cli_3.6.1                     Biobase_2.58.0               </span></span>
<span><span class="co">#&gt; [23] DelayedArray_0.24.0           desc_1.4.2                   </span></span>
<span><span class="co">#&gt; [25] labeling_0.4.2                bookdown_0.33                </span></span>
<span><span class="co">#&gt; [27] sass_0.4.5                    scales_1.2.1                 </span></span>
<span><span class="co">#&gt; [29] rappdirs_0.3.3                pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt; [31] Rsamtools_2.14.0              systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [33] digest_0.6.31                 rmarkdown_2.21               </span></span>
<span><span class="co">#&gt; [35] XVector_0.38.0                pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt; [37] htmltools_0.5.5               MatrixGenerics_1.10.0        </span></span>
<span><span class="co">#&gt; [39] fastmap_1.1.1                 highr_0.10                   </span></span>
<span><span class="co">#&gt; [41] rlang_1.1.0                   RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt; [43] shiny_1.7.4                   BiocIO_1.8.0                 </span></span>
<span><span class="co">#&gt; [45] jquerylib_0.1.4               generics_0.1.3               </span></span>
<span><span class="co">#&gt; [47] farver_2.1.1                  jsonlite_1.8.4               </span></span>
<span><span class="co">#&gt; [49] BiocParallel_1.32.6           RCurl_1.98-1.12              </span></span>
<span><span class="co">#&gt; [51] magrittr_2.0.3                GenomeInfoDbData_1.2.9       </span></span>
<span><span class="co">#&gt; [53] Matrix_1.5-4                  Rcpp_1.0.10                  </span></span>
<span><span class="co">#&gt; [55] munsell_0.5.0                 fansi_1.0.4                  </span></span>
<span><span class="co">#&gt; [57] lifecycle_1.0.3               stringi_1.7.12               </span></span>
<span><span class="co">#&gt; [59] yaml_2.3.7                    SummarizedExperiment_1.28.0  </span></span>
<span><span class="co">#&gt; [61] zlibbioc_1.44.0               grid_4.2.3                   </span></span>
<span><span class="co">#&gt; [63] blob_1.2.4                    parallel_4.2.3               </span></span>
<span><span class="co">#&gt; [65] promises_1.2.0.1              crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [67] lattice_0.21-8                Biostrings_2.66.0            </span></span>
<span><span class="co">#&gt; [69] excluderanges_0.99.10         KEGGREST_1.38.0              </span></span>
<span><span class="co">#&gt; [71] knitr_1.42                    pillar_1.9.0                 </span></span>
<span><span class="co">#&gt; [73] rjson_0.2.21                  codetools_0.2-19             </span></span>
<span><span class="co">#&gt; [75] XML_3.99-0.14                 glue_1.6.2                   </span></span>
<span><span class="co">#&gt; [77] BiocVersion_3.16.0            evaluate_0.20                </span></span>
<span><span class="co">#&gt; [79] BiocManager_1.30.20           png_0.1-8                    </span></span>
<span><span class="co">#&gt; [81] vctrs_0.6.2                   httpuv_1.6.9                 </span></span>
<span><span class="co">#&gt; [83] gtable_0.3.3                  purrr_1.0.1                  </span></span>
<span><span class="co">#&gt; [85] cachem_1.0.7                  xfun_0.38                    </span></span>
<span><span class="co">#&gt; [87] mime_0.12                     xtable_1.8-4                 </span></span>
<span><span class="co">#&gt; [89] restfulr_0.0.15               later_1.3.0                  </span></span>
<span><span class="co">#&gt; [91] ragg_1.2.5                    tibble_3.2.1                 </span></span>
<span><span class="co">#&gt; [93] GenomicAlignments_1.34.1      AnnotationDbi_1.60.2         </span></span>
<span><span class="co">#&gt; [95] memoise_2.0.1                 ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [97] interactiveDisplayBase_1.36.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Amemiya:2019aa" class="csl-entry">
Amemiya, Haley M, Anshul Kundaje, and Alan P Boyle. 2019. <span>“The
ENCODE Blacklist: Identification of Problematic Regions of the
Genome.”</span> <em>Sci Rep</em> 9 (1): 9354. <a href="https://doi.org/10.1038/s41598-019-45839-z" class="external-link">https://doi.org/10.1038/s41598-019-45839-z</a>.
</div>
<div id="ref-Klasfeld2022-nn" class="csl-entry">
Klasfeld, Sammy, and Doris Wagner. 2022. <span>“Greenscreen Decreases
Type <span>I</span> Errors and Increases True Peak Detection in Genomic
Datasets Including <span class="nocase">ChIP-seq</span>.”</span>
<em>bioRxiv</em>.
</div>
<div id="ref-nordin2022cut" class="csl-entry">
Nordin, Anna, Gianluca Zambanini, Pierfrancesco Pagella, and Claudio
Cantu. 2022. <span>“The CUT&amp;RUN Blacklist of Problematic Regions of
the Genome.”</span> <em>bioRxiv</em>.
</div>
<div id="ref-Wimberley:2019ub" class="csl-entry">
Wimberley, Charles E, and Steffen Heber. 2019. <span>“PeakPass:
Automating ChIP-Seq Blacklist Creation.”</span> <em>J Comput Biol</em>,
December. <a href="https://doi.org/10.1089/cmb.2019.0295" class="external-link">https://doi.org/10.1089/cmb.2019.0295</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Mikhail Dozmorov, Eric Davis, Wancen Mu, Stuart Lee, Tim Triche, Douglas Phanstiel, Michael Love.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
