<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to excluderanges • excluderanges</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to excluderanges">
<meta property="og:description" content="excluderanges">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">excluderanges</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/excluderanges.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/mdozmorov/excluderanges/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="excluderanges_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to excluderanges</h1>
                        <h4 class="author">Mikhail Dozmorov</h4>
            <address class="author_afil">
      Virginia Commonwealth University<br><a class="author_email" href="mailto:#"></a><a href="mailto:mikhail.dozmorov@gmail.com" class="email">mikhail.dozmorov@gmail.com</a>
      </address>
                  
            <h4 class="date">31 August 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mdozmorov/excluderanges/blob/master/vignettes/excluderanges.Rmd"><code>vignettes/excluderanges.Rmd</code></a></small>
      <div class="hidden name"><code>excluderanges.Rmd</code></div>

    </div>

    
    
<div id="exclude-ranges" class="section level1">
<h1 class="hasAnchor">
<a href="#exclude-ranges" class="anchor"></a>Exclude ranges</h1>
<p>Genomic ranges of problematic genomic regions that should be avoided when working with genomic data. For human, mouse, and selected model organisms.</p>
<!-- badges: start -->
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a> <!-- badges: end --></p>
<p>TL;DR - For human hg38 genome assembly, <a href="https://twitter.com/anshulkundaje">Anshul</a> <a href="https://twitter.com/anshulkundaje/status/1263546023151992832?s=20">recommends</a> <a href="https://www.encodeproject.org/files/ENCFF356LFX/">ENCFF356LFX exclusion list regions</a>.</p>
<p>BED files of exclusion regions are available on the <a href="https://www.encodeproject.org/search/?searchTerm=exclusion+list">ENCODE project</a> website <span class="citation">(Amemiya, Kundaje, and Boyle 2019)</span>. Human (hg19, hg38) and mouse (mm9, mm10) exclusion regions are available. However, exclusion lists generated by multiple labs often create uncertainty what to use. The purpose of this package is to provide a unified place for informed retrieval of exclusion regions.</p>
<p>Naming convention: <code>&lt;genome assembly&gt;.&lt;lab&gt;.&lt;original file name&gt;</code>, e.g., <code>hg19.Birney.wgEncodeDacMapabilityConsensusExcludable</code>.</p>
<p>Download all data from the <a href="https://drive.google.com/drive/folders/1MFn6LPZD1zZRQz7biR0Fyl-mzkx4CIxS?usp=sharing">Google Drive folder</a></p>
<p>See <a href="inst/scripts/make-data.R">make-data.R</a> how to create the excluderanges GRanges objects.</p>
<div id="install-excluderanges" class="section level2">
<h2 class="hasAnchor">
<a href="#install-excluderanges" class="anchor"></a>Install <code>excluderanges</code>
</h2>
<!--`R` is an open-source statistical environment which can be easily modified to enhance its functionality via packages. *[excluderanges](https://bioconductor.org/packages/3.13/excluderanges)* is a `R` package available via the [Bioconductor](http://bioconductor.org) repository for packages. `R` can be installed on any operating system from [CRAN](https://cran.r-project.org/) after which you can install *[excluderanges](https://bioconductor.org/packages/3.13/excluderanges)* by using the following commands in your `R` session:-->
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># Install the development version of Bioconductor (need 3.14 and above)</span>
<span class="co"># BiocManager::install(version = "devel")</span>
<span class="co"># Check that you have a valid Bioconductor installation</span>
<span class="co"># BiocManager::valid()</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"mdozmorov/excluderanges"</span><span class="op">)</span></code></pre></div>
</div>
<div id="use-excluderanges" class="section level2">
<h2 class="hasAnchor">
<a href="#use-excluderanges" class="anchor"></a>Use excluderanges</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># hg38 excluderanges coordinates</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=1Jy_7cvFj5ZHUUwHgGUKZeFk8lFCalpKk"</span>, destfile <span class="op">=</span> <span class="st">"hg38.Kundaje.GRCh38_unified_Excludable.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Kundaje.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg38.Kundaje.GRCh38_unified_Excludable.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Kundaje.1</span></code></pre></div>
<pre><code>&gt; excludeGR.hg38.Kundaje.1
GRanges object with 910 ranges and 0 metadata columns:
        seqnames            ranges strand
           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
    [1]     chr1     628903-635104      *
    [2]     chr1   5850087-5850571      *
    [3]     chr1   8909610-8910014      *
    [4]     chr1   9574580-9574997      *
    [5]     chr1 32043823-32044203      *
    ...      ...               ...    ...
  [906]     chrY 11290797-11334278      *
  [907]     chrY 11493053-11592850      *
  [908]     chrY 11671014-11671046      *
  [909]     chrY 11721528-11749472      *
  [910]     chrY 56694632-56889743      *
  -------
  seqinfo: 24 sequences from hg38 genome</code></pre>
<p>We can load other excludable regions for the hg38 genome assembly and compare them.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=13RQmvqwC9qaYE65i74U1N-Rb4R-Sy2ud"</span>, destfile <span class="op">=</span> <span class="st">"hg38.Bernstein.Mint_Excludable_GRCh38.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Bernstein</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg38.Bernstein.Mint_Excludable_GRCh38.rds"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=1ULI6uvs0vTlPcZ5yyLdsSgX1mIqnnlpp"</span>, destfile <span class="op">=</span> <span class="st">"hg38.Kundaje.GRCh38.Excludable.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Kundaje.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg38.Kundaje.GRCh38.Excludable.rds"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=1TA5Y8nmutoAyhL8XEl2qQTnmW6gl8CAg"</span>, destfile <span class="op">=</span> <span class="st">"hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Reddy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=1psVWkiaLy4nxAWssqv5dEZyNUrDIm005"</span>, destfile <span class="op">=</span> <span class="st">"hg38.Wold.hg38mitoExcludable.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Wold</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg38.Wold.hg38mitoExcludable.rds"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=1pwbBEOHFfnVb6I9UljF3bJQLF3et4WV7"</span>, destfile <span class="op">=</span> <span class="st">"hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds"</span><span class="op">)</span>
<span class="va">excludeGR.hg38.Yeo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds"</span><span class="op">)</span></code></pre></div>
<p>Compare the number of excludable regions.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span>, 
                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span>, 
                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span>, 
                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span>, 
                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span>, 
                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span>,
                          Source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, 
                                     <span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, 
                                     <span class="st">"Kundaje.GRCh38.Excludable"</span>, 
                                     <span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, 
                                     <span class="st">"Wold.hg38mitoExcludable"</span>, 
                                     <span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover.bed"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Order Source by the number of regions</span>
<span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span>, levels <span class="op">=</span> <span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Source</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Count</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Source</span>, y <span class="op">=</span> <span class="va">Count</span>, fill <span class="op">=</span> <span class="va">Source</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"../man/figures/excluderanges_hg38_count.png"</span>, width <span class="op">=</span> <span class="fl">5.5</span>, height <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/excluderanges_hg38_count.png" width="825"></p>
<p>Compare the width of excludable regions. log2 scale because of heavy right tail distributions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggridges/">ggridges</a></span><span class="op">)</span>
<span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span>, 
                                    <span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span>, 
                                    <span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span>, 
                                    <span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span>, 
                                    <span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span>, 
                                    <span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span>,
                          Source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Kundaje.GRCh38.Excludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Wold.hg38mitoExcludable"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span><span class="op">)</span>,
                                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover.bed"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="va">Width</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">Source</span>, fill <span class="op">=</span> <span class="va">Source</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://wilkelab.org/ggridges/reference/geom_density_ridges.html">geom_density_ridges</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"../man/figures/excluderanges_hg38_width.png"</span>, width <span class="op">=</span> <span class="fl">5.5</span>, height <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/excluderanges_hg38_width.png" width="825"></p>
<p>We can investigate the total width of each set of excludable ranges.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>TotalWidth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span><span class="op">)</span><span class="op">)</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.1</span><span class="op">)</span><span class="op">)</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Kundaje.2</span><span class="op">)</span><span class="op">)</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span><span class="op">)</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Wold</span><span class="op">)</span><span class="op">)</span>, 
                                         <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, 
                          Source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, 
                                     <span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, 
                                     <span class="st">"Kundaje.GRCh38.Excludable"</span>, 
                                     <span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, 
                                     <span class="st">"Wold.hg38mitoExcludable"</span>, 
                                     <span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">TotalWidth</span>, y <span class="op">=</span> <span class="va">Source</span>, fill <span class="op">=</span> <span class="va">Source</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_discrete.html">scale_y_discrete</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">abbreviate</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"log10 total width"</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"../man/figures/excluderanges_hg38_sumwidth.png"</span>, width <span class="op">=</span> <span class="fl">6.5</span>, height <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/excluderanges_hg38_sumwidth.png" width="975"></p>
<p>We can compare Jaccard overlap between those sets of excludable regions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
<span class="co"># Jaccard calculations</span>
<span class="va">jaccard</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr_a</span>, <span class="va">gr_b</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">intersects</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">intersect</a></span><span class="op">(</span><span class="va">gr_a</span>, <span class="va">gr_b</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
  <span class="va">intersection</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">intersects</span><span class="op">)</span><span class="op">)</span>
  <span class="va">union</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/setops-methods.html">union</a></span><span class="op">(</span><span class="va">gr_a</span>, <span class="va">gr_b</span>, ignore.strand <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">DataFrame</span><span class="op">(</span><span class="va">intersection</span>, <span class="va">union</span>, 
            jaccard <span class="op">=</span> <span class="va">intersection</span><span class="op">/</span><span class="va">union</span>,
             n_intersections <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">intersects</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># List and names of all excludable regions</span>
<span class="va">all_excludeGR_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span>, 
                        <span class="va">excludeGR.hg38.Kundaje.1</span>, 
                        <span class="va">excludeGR.hg38.Kundaje.2</span>,
                        <span class="va">excludeGR.hg38.Reddy</span>,
                        <span class="va">excludeGR.hg38.Wold</span>,
                        <span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span>
<span class="va">all_excludeGR_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Bernstein.Mint_Excludable_GRCh38"</span>, 
                     <span class="st">"Kundaje.GRCh38_unified_Excludable"</span>, 
                     <span class="st">"Kundaje.GRCh38.Excludable"</span>, 
                     <span class="st">"Reddy.wgEncodeDacMapabilityConsensusExcludable"</span>, 
                     <span class="st">"Wold.hg38mitoExcludable"</span>, 
                     <span class="st">"Yeo.eCLIP_Excludableregions.hg38liftover"</span><span class="op">)</span>
<span class="co"># Correlation matrix, empty</span>
<span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span>data <span class="op">=</span> <span class="fl">0</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Fill it in</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># If diagonal, set to zero</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">i</span> <span class="op">==</span> <span class="va">j</span><span class="op">)</span> <span class="va">mtx_to_plot</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>
    <span class="co"># Process only one half, the other is symmetric</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">i</span> <span class="op">&gt;</span> <span class="va">j</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">mtx_to_plot</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mtx_to_plot</span><span class="op">[</span><span class="va">j</span>, <span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">jaccard</span><span class="op">(</span><span class="va">all_excludeGR_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, <span class="va">all_excludeGR_list</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="st">"jaccard"</span><span class="op">]</span><span class="op">]</span>
    <span class="op">}</span>
  <span class="op">}</span>
<span class="op">}</span>
<span class="co"># Trim row/colnames</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_trunc.html">str_trunc</a></span><span class="op">(</span><span class="va">all_excludeGR_name</span>, width <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> 
<span class="co"># Save the plot</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/png.html">png</a></span><span class="op">(</span><span class="st">"../man/figures/excluderanges_hg38_jaccard.png"</span>, width <span class="op">=</span> <span class="fl">1000</span>, height <span class="op">=</span> <span class="fl">900</span>, res <span class="op">=</span> <span class="fl">200</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.matrix.html">data.matrix</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/excluderanges_hg38_jaccard.png" width="500"></p>
<p>Note that some excludable ranges objects contain six columns, implying there may be some interesting metadata. Let’s explore one.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">mcols</span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span>
<span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">excludeGR.hg38.Reddy</span><span class="op">)</span><span class="op">[[</span><span class="st">"name"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Type"</span>, <span class="st">"Number"</span><span class="op">)</span>
<span class="va">mtx_to_plot</span> <span class="op">&lt;-</span> <span class="va">mtx_to_plot</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Number</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Type</span>, levels <span class="op">=</span> <span class="va">mtx_to_plot</span><span class="op">$</span><span class="va">Type</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">mtx_to_plot</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Number</span>, y <span class="op">=</span> <span class="va">Type</span>, fill <span class="op">=</span> <span class="va">Type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"../man/figures/excluderanges_hg38_Reddy_metadata.png"</span>, width <span class="op">=</span> <span class="fl">5</span>, height <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></code></pre></div>
<p><img src="../man/figures/excluderanges_hg38_Reddy_metadata.png" width="750"></p>
<p>One may decide to combine the excludable ranges from all labs, although from previous results we may decide to follow Anshul’s <a href="https://twitter.com/anshulkundaje/status/1263546023151992832?s=20">advice</a> advice about the <a href="https://www.encodeproject.org/files/ENCFF356LFX/">ENCFF356LFX exclusion list regions</a> and use the <code>excludeGR.hg38.Kundaje.1</code> object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">excludeGR.hg38.all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/ggridges/reference/reduce.html">reduce</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">excludeGR.hg38.Bernstein</span>, <span class="va">excludeGR.hg38.Kundaje.1</span>, <span class="va">excludeGR.hg38.Kundaje.2</span>, <span class="va">excludeGR.hg38.Reddy</span>, <span class="va">excludeGR.hg38.Wold</span>, <span class="va">excludeGR.hg38.Yeo</span><span class="op">)</span><span class="op">)</span>
<span class="co"># Keep only standard chromosomes</span>
<span class="va">excludeGR.hg38.all</span> <span class="op">&lt;-</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">excludeGR.hg38.all</span>, pruning.mode <span class="op">=</span> <span class="st">"coarse"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">excludeGR.hg38.all</span><span class="op">)</span><span class="op">)</span>
<span class="co"># [1] 13239</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">excludeGR.hg38.all</span><span class="op">)</span><span class="op">)</span>
<span class="co">#    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#       5    1778    2306    8153    2859 5407757 </span></code></pre></div>
</div>
<div id="centromeres-telomeres-etc-" class="section level2">
<h2 class="hasAnchor">
<a href="#centromeres-telomeres-etc-" class="anchor"></a>Centromeres, telomeres, etc.</h2>
<p>Besides the ENCODE-produced excludable regions, we may want to exclude centromeres, telomeres, and other gap locations. The “Gap Locations” track for Homo Sapiens is available for the GRcH37/hg19 genome assembly as a <a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema">UCSC ‘gap’ table</a>. It can be retrieved from <em><a href="https://bioconductor.org/packages/3.13/AnnotationHub">AnnotationHub</a></em>, but lacks the metadata columns needed to decide the type of gaps.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span><span class="op">)</span>
<span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># Search for the gap track</span>
<span class="co"># ahData &lt;- query(ah, c("gap", "Homo sapiens", "hg19"))</span>
<span class="co"># ahData[ctcfData$title == "Gap"]</span>
<span class="va">gaps</span> <span class="op">&lt;-</span> <span class="va">ahData</span><span class="op">[[</span><span class="st">"AH6444"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code>&gt; gaps
UCSC track 'gap'
UCSCData object with 457 ranges and 0 metadata columns:
               seqnames              ranges strand
                  &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;
    [1]            chr1 124535435-142535434      *
    [2]            chr1 121535435-124535434      *
    [3]            chr1     3845269-3995268      *
    [4]            chr1   13219913-13319912      *
    [5]            chr1   17125659-17175658      *
    ...             ...                 ...    ...
  [453]  chr6_ssto_hap7     4639807-4661556      *
  [454]  chr6_ssto_hap7     4723510-4774367      *
  [455]  chr6_ssto_hap7     4783799-4836049      *
  [456] chr17_ctg5_hap1     1256795-1306794      *
  [457] chr17_ctg5_hap1     1588969-1638968      *
  -------
  seqinfo: 93 sequences (1 circular) from hg19 genome</code></pre>
<p>The <a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema">UCSC ‘gap’ table</a> provides better granularity about the types of gaps available. E.g., for human, hg19, we have the following types and the number of gaps.</p>
<p><img src="../man/figures/excluderanges_hg19_gaps_number.png" width="750"></p>
<p>Those objects are provided as individual GRanges.</p>
<p>Naming convention: <code>&lt;genome assembly&gt;.UCSC.&lt;gap type&gt;</code>, e.g., <code>hg19.UCSC.gap_centromere</code>.</p>
<p>Download all data from the <a href="https://drive.google.com/drive/folders/1MFn6LPZD1zZRQz7biR0Fyl-mzkx4CIxS?usp=sharing">Google Drive folder</a></p>
<p>We can similarly load any gap type object.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://drive.google.com/uc?export=download&amp;id=1uno7XzgCdFJeMjYI6GMfU5bAwEBWaUlR"</span>, destfile <span class="op">=</span> <span class="st">"hg19.UCSC.centromere.rds"</span><span class="op">)</span>
<span class="va">gapsGR_hg19_centromere</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"hg19.UCSC.centromere.rds"</span><span class="op">)</span>
<span class="va">gapsGR_hg19_centromere</span></code></pre></div>
<pre><code>&gt; gapsGR_hg19_centromere
GRanges object with 24 ranges and 6 metadata columns:
      seqnames              ranges strand |       bin        ix           n      size        type      bridge
         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;character&gt; &lt;numeric&gt; &lt;character&gt; &lt;character&gt;
    2     chr1 121535434-124535434      * |        23      1270           N   3000000  centromere          no
  184    chr21   11288129-14288129      * |        10        22           N   3000000  centromere          no
  199    chr22   13000000-16000000      * |        10         3           N   3000000  centromere          no
  206    chr19   24681782-27681782      * |         1       410           N   3000000  centromere          no
  224     chrY   10104553-13104553      * |        10       105           N   3000000  centromere          no
  ...      ...                 ...    ... .       ...       ...         ...       ...         ...         ...
  439     chr6   58830166-61830166      * |        16       628           N   3000000  centromere          no
  453     chr5   46405641-49405641      * |        14       452           N   3000000  centromere          no
  460     chr4   49660117-52660117      * |         1       447           N   3000000  centromere          no
  476     chr3   90504854-93504854      * |         2       784           N   3000000  centromere          no
  481     chr2   92326171-95326171      * |        20       770           N   3000000  centromere          no
  -------
  seqinfo: 24 sequences from hg19 genome</code></pre>
<div id="centromeres-for-the-hg38-genome-assembly" class="section level3">
<h3 class="hasAnchor">
<a href="#centromeres-for-the-hg38-genome-assembly" class="anchor"></a>Centromeres for the hg38 genome assembly</h3>
<p>Note that the UCSC ‘gap’ table for the hg38 human genome assembly does not contain genomic coordinates for the “centromere” gap type. These can be obtained from the <em><a href="https://bioconductor.org/packages/3.13/rCGH">rCGH</a></em> package as follows:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/fredcommo/rCGH">rCGH</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span><span class="op">)</span>
<span class="co"># hg38 # data.frame</span>
<span class="co"># Adjust chromosome names</span>
<span class="va">hg38</span><span class="op">$</span><span class="va">chrom</span><span class="op">[</span><span class="va">hg38</span><span class="op">$</span><span class="va">chrom</span> <span class="op">==</span> <span class="fl">23</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"X"</span>
<span class="va">hg38</span><span class="op">$</span><span class="va">chrom</span><span class="op">[</span><span class="va">hg38</span><span class="op">$</span><span class="va">chrom</span> <span class="op">==</span> <span class="fl">24</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Y"</span>
<span class="va">hg38</span><span class="op">$</span><span class="va">chrom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="va">hg38</span><span class="op">$</span><span class="va">chrom</span><span class="op">)</span>
<span class="co"># Make GRanges object</span>
<span class="va">hg38.UCSC.centromere</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">hg38</span>, seqnames.field <span class="op">=</span> <span class="st">"chrom"</span>, start.field <span class="op">=</span> <span class="st">"centromerStart"</span>, end.field <span class="op">=</span> <span class="st">"centromerEnd"</span><span class="op">)</span>
<span class="co"># Assign seqinfo data</span>
<span class="fu">seqlengths</span><span class="op">(</span><span class="va">hg38.UCSC.centromere</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">hg38</span><span class="op">$</span><span class="va">length</span>
<span class="fu">genome</span><span class="op">(</span><span class="va">hg38.UCSC.centromere</span><span class="op">)</span>     <span class="op">&lt;-</span> <span class="st">"hg38"</span>
<span class="co"># Resulting object</span>
<span class="va">hg38.UCSC.centromere</span>
<span class="co">#&gt; GRanges object with 24 ranges and 0 metadata columns:</span>
<span class="co">#&gt;        seqnames              ranges strand</span>
<span class="co">#&gt;           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;    [1]     chr1 121535434-124535434      *</span>
<span class="co">#&gt;    [2]     chr2   92326171-95326171      *</span>
<span class="co">#&gt;    [3]     chr3   90504854-93504854      *</span>
<span class="co">#&gt;    [4]     chr4   49660117-52660117      *</span>
<span class="co">#&gt;    [5]     chr5   46405641-49405641      *</span>
<span class="co">#&gt;    ...      ...                 ...    ...</span>
<span class="co">#&gt;   [20]    chr20   26369569-29369569      *</span>
<span class="co">#&gt;   [21]    chr21   11288129-14288129      *</span>
<span class="co">#&gt;   [22]    chr22   13000000-16000000      *</span>
<span class="co">#&gt;   [23]     chrX   58632012-61632012      *</span>
<span class="co">#&gt;   [24]     chrY   10104553-13104553      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from hg38 genome</span></code></pre></div>
<p>The <em><a href="https://bioconductor.org/packages/3.13/rCGH">rCGH</a></em> package also contains data for the <code>hg19</code> and <code>hg18</code> genomes. The <code>hg19</code> centromere data is equivalent to the <code>hg19.UCSC.centromere</code> object provided in our <em><a href="https://bioconductor.org/packages/3.13/excluderanges">excluderanges</a></em> package.</p>
</div>
</div>
<div id="source-data-for-the-excludable-regions" class="section level2">
<h2 class="hasAnchor">
<a href="#source-data-for-the-excludable-regions" class="anchor"></a>Source data for the excludable regions</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"../man/table_excluderanges.csv"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">mtx</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="28%">
<col width="8%">
<col width="4%">
<col width="11%">
<col width="8%">
<col width="38%">
</colgroup>
<thead><tr class="header">
<th align="left">Object</th>
<th align="right">Number.of.regions</th>
<th align="left">Assembly</th>
<th align="left">Lab</th>
<th align="right">Number.of.columns</th>
<th align="left">Source</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ce10.Kundaje.ce10-Excludable.rds</td>
<td align="right">122</td>
<td align="left">ce10</td>
<td align="left">Anshul Kundaje, Stanford</td>
<td align="right">3</td>
<td align="left"><a href="http://mitra.stanford.edu/kundaje/akundaje/release/Excludables/ce10-C.elegans" class="uri">http://mitra.stanford.edu/kundaje/akundaje/release/Excludables/ce10-C.elegans</a></td>
</tr>
<tr class="even">
<td align="left">dm3.Kundaje.dm3-Excludable.rds</td>
<td align="right">492</td>
<td align="left">dm3</td>
<td align="left">Anshul Kundaje, Stanford</td>
<td align="right">3</td>
<td align="left"><a href="http://mitra.stanford.edu/kundaje/akundaje/release/Excludables/dm3-D.melanogaster/" class="uri">http://mitra.stanford.edu/kundaje/akundaje/release/Excludables/dm3-D.melanogaster/</a></td>
</tr>
<tr class="odd">
<td align="left">hg19.Bernstein.Mint_Excludable_hg19.rds</td>
<td align="right">9035</td>
<td align="left">hg19</td>
<td align="left">Bradley Bernstein, Broad</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF200UUD/" class="uri">https://www.encodeproject.org/files/ENCFF200UUD/</a></td>
</tr>
<tr class="even">
<td align="left">hg19.Birney.wgEncodeDacMapabilityConsensusExcludable.rds</td>
<td align="right">411</td>
<td align="left">hg19</td>
<td align="left">Ewan Birney, EBI</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF001TDO/" class="uri">https://www.encodeproject.org/files/ENCFF001TDO/</a></td>
</tr>
<tr class="odd">
<td align="left">hg19.Crawford.wgEncodeDukeMapabilityRegionsExcludable.rds</td>
<td align="right">1649</td>
<td align="left">hg19</td>
<td align="left">Gregory Crawford, Duke</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF001THR/" class="uri">https://www.encodeproject.org/files/ENCFF001THR/</a></td>
</tr>
<tr class="even">
<td align="left">hg19.Wold.hg19mitoExcludable.rds</td>
<td align="right">295</td>
<td align="left">hg19</td>
<td align="left">Barbara Wold, Caltech</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF055QTV/" class="uri">https://www.encodeproject.org/files/ENCFF055QTV/</a></td>
</tr>
<tr class="odd">
<td align="left">hg19.Yeo.eCLIP_Excludableregions.hg19.rds</td>
<td align="right">57</td>
<td align="left">hg19</td>
<td align="left">Gene Yeo, UCSD</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF039QTN/" class="uri">https://www.encodeproject.org/files/ENCFF039QTN/</a></td>
</tr>
<tr class="even">
<td align="left">hg38.Bernstein.Mint_Excludable_GRCh38.rds</td>
<td align="right">12052</td>
<td align="left">hg38</td>
<td align="left">Bradley Bernstein, Broad</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF023CZC/" class="uri">https://www.encodeproject.org/files/ENCFF023CZC/</a></td>
</tr>
<tr class="odd">
<td align="left">hg38.Kundaje.GRCh38.Excludable.rds</td>
<td align="right">38</td>
<td align="left">hg38</td>
<td align="left">Anshul Kundaje, Stanford</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF356LFX/" class="uri">https://www.encodeproject.org/files/ENCFF356LFX/</a></td>
</tr>
<tr class="even">
<td align="left">hg38.Kundaje.GRCh38_unified_Excludable.rds</td>
<td align="right">910</td>
<td align="left">hg38</td>
<td align="left">Anshul Kundaje, Stanford</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF419RSJ/" class="uri">https://www.encodeproject.org/files/ENCFF419RSJ/</a></td>
</tr>
<tr class="odd">
<td align="left">hg38.Reddy.wgEncodeDacMapabilityConsensusExcludable.hg38.rds</td>
<td align="right">401</td>
<td align="left">hg38</td>
<td align="left">Tim Reddy, Duke</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF220FIN/" class="uri">https://www.encodeproject.org/files/ENCFF220FIN/</a></td>
</tr>
<tr class="even">
<td align="left">hg38.Wold.hg38mitoExcludable.rds</td>
<td align="right">299</td>
<td align="left">hg38</td>
<td align="left">Barbara Wold, Caltech</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF940NTE/" class="uri">https://www.encodeproject.org/files/ENCFF940NTE/</a></td>
</tr>
<tr class="odd">
<td align="left">hg38.Yeo.eCLIP_Excludableregions.hg38liftover.bed.fixed.rds</td>
<td align="right">56</td>
<td align="left">hg38</td>
<td align="left">Gene Yeo, UCSD</td>
<td align="right">6</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF269URO/" class="uri">https://www.encodeproject.org/files/ENCFF269URO/</a></td>
</tr>
<tr class="even">
<td align="left">mm10.Hardison.Excludable.full.rds</td>
<td align="right">7865</td>
<td align="left">mm10</td>
<td align="left">Ross Hardison, PennState</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF790DJT/" class="uri">https://www.encodeproject.org/files/ENCFF790DJT/</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.Hardison.psuExcludable.mm10.rds</td>
<td align="right">5552</td>
<td align="left">mm10</td>
<td align="left">Ross Hardison, PennState</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF226BDM/" class="uri">https://www.encodeproject.org/files/ENCFF226BDM/</a></td>
</tr>
<tr class="even">
<td align="left">mm10.Kundaje.anshul.Excludable.mm10.rds</td>
<td align="right">3010</td>
<td align="left">mm10</td>
<td align="left">Anshul Kundaje, Stanford</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF999QPV/" class="uri">https://www.encodeproject.org/files/ENCFF999QPV/</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.Kundaje.mm10.Excludable.rds</td>
<td align="right">164</td>
<td align="left">mm10</td>
<td align="left">Anshul Kundaje, Stanford</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF547MET/" class="uri">https://www.encodeproject.org/files/ENCFF547MET/</a></td>
</tr>
<tr class="even">
<td align="left">mm10.Wold.mm10mitoExcludable.rds</td>
<td align="right">123</td>
<td align="left">mm10</td>
<td align="left">Barbara Wold, Caltech</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF759PJK/" class="uri">https://www.encodeproject.org/files/ENCFF759PJK/</a></td>
</tr>
<tr class="odd">
<td align="left">mm9.Wold.mm9mitoExcludable.rds</td>
<td align="right">123</td>
<td align="left">mm9</td>
<td align="left">Barbara Wold, Caltech</td>
<td align="right">3</td>
<td align="left"><a href="https://www.encodeproject.org/files/ENCFF299EZH/" class="uri">https://www.encodeproject.org/files/ENCFF299EZH/</a></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mtx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="st">"../man/table_gap.csv"</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">mtx</span><span class="op">)</span></code></pre></div>
<table class="table">
<colgroup>
<col width="14%">
<col width="8%">
<col width="4%">
<col width="2%">
<col width="8%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th align="left">Object</th>
<th align="right">Number.of.regions</th>
<th align="left">Assembly</th>
<th align="left">Lab</th>
<th align="right">Number.of.columns</th>
<th align="left">Source</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">hg19.UCSC.centromere.rds</td>
<td align="right">24</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.clone.rds</td>
<td align="right">207</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.contig.rds</td>
<td align="right">163</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.heterochromatin.rds</td>
<td align="right">12</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.scaffold.rds</td>
<td align="right">40</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">hg19.UCSC.short_arm.rds</td>
<td align="right">5</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">hg19.UCSC.telomere.rds</td>
<td align="right">46</td>
<td align="left">hg19</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg19&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">hg38.UCSC.contig.rds</td>
<td align="right">285</td>
<td align="left">hg38</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">hg38.UCSC.heterochromatin.rds</td>
<td align="right">11</td>
<td align="left">hg38</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">hg38.UCSC.scaffold.rds</td>
<td align="right">478</td>
<td align="left">hg38</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">hg38.UCSC.short_arm.rds</td>
<td align="right">5</td>
<td align="left">hg38</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">hg38.UCSC.telomere.rds</td>
<td align="right">48</td>
<td align="left">hg38</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=hg38&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.centromere.rds</td>
<td align="right">20</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.clone.rds</td>
<td align="right">114</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.contig.rds</td>
<td align="right">104</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.fragment.rds</td>
<td align="right">1</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.other.rds</td>
<td align="right">384</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">mm10.UCSC.short_arm.rds</td>
<td align="right">21</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.telomere.rds</td>
<td align="right">42</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">mm9.UCSC.centromere.rds</td>
<td align="right">21</td>
<td align="left">mm9</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm9&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm9&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">mm9.UCSC.contig.rds</td>
<td align="right">281</td>
<td align="left">mm9</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm9&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm9&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="even">
<td align="left">mm9.UCSC.fragment.rds</td>
<td align="right">709</td>
<td align="left">mm9</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm9&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm9&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
<tr class="odd">
<td align="left">mm10.UCSC.scaffold.rds</td>
<td align="right">48</td>
<td align="left">mm10</td>
<td align="left">UCSC</td>
<td align="right">9</td>
<td align="left"><a href="http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables?db=mm10&amp;hgta_group=map&amp;hgta_track=gap&amp;hgta_table=gap&amp;hgta_doSchema=describe+table+schema</a></td>
</tr>
</tbody>
</table>
</div>
<div id="r-session-information-" class="section level2">
<h2 class="hasAnchor">
<a href="#r-session-information-" class="anchor"></a><code>R</code> session information.</h2>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.1.0 (2021-05-18)
#&gt;  os       macOS Catalina 10.15.7      
#&gt;  system   x86_64, darwin17.0          
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       America/New_York            
#&gt;  date     2021-08-31                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package                           * version  date       lib source        
#&gt;  aCGH                                1.70.0   2021-05-19 [1] Bioconductor  
#&gt;  affy                                1.70.0   2021-05-19 [1] Bioconductor  
#&gt;  affyio                              1.62.0   2021-05-19 [1] Bioconductor  
#&gt;  AnnotationDbi                       1.54.1   2021-06-08 [1] Bioconductor  
#&gt;  assertthat                          0.2.1    2019-03-21 [1] CRAN (R 4.1.0)
#&gt;  Biobase                             2.52.0   2021-05-19 [1] Bioconductor  
#&gt;  BiocFileCache                       2.0.0    2021-05-19 [2] Bioconductor  
#&gt;  BiocGenerics                      * 0.38.0   2021-05-19 [1] Bioconductor  
#&gt;  BiocIO                              1.2.0    2021-05-19 [1] Bioconductor  
#&gt;  BiocManager                         1.30.16  2021-06-15 [1] CRAN (R 4.1.0)
#&gt;  BiocParallel                        1.26.2   2021-08-22 [1] Bioconductor  
#&gt;  BiocStyle                         * 2.20.2   2021-06-17 [1] Bioconductor  
#&gt;  biomaRt                             2.48.2   2021-07-01 [2] Bioconductor  
#&gt;  Biostrings                          2.60.2   2021-08-05 [1] Bioconductor  
#&gt;  bit                                 4.0.4    2020-08-04 [1] CRAN (R 4.1.0)
#&gt;  bit64                               4.0.5    2020-08-30 [1] CRAN (R 4.1.0)
#&gt;  bitops                              1.0-7    2021-04-24 [1] CRAN (R 4.1.0)
#&gt;  blob                                1.2.2    2021-07-23 [1] CRAN (R 4.1.0)
#&gt;  bookdown                            0.23     2021-08-13 [1] CRAN (R 4.1.0)
#&gt;  bslib                               0.2.5.1  2021-05-18 [1] CRAN (R 4.1.0)
#&gt;  cachem                              1.0.5    2021-05-15 [2] CRAN (R 4.1.0)
#&gt;  cli                                 3.0.1    2021-07-17 [1] CRAN (R 4.1.0)
#&gt;  cluster                             2.1.2    2021-04-17 [2] CRAN (R 4.1.0)
#&gt;  colorspace                          2.0-2    2021-06-24 [1] CRAN (R 4.1.0)
#&gt;  crayon                              1.4.1    2021-02-08 [2] CRAN (R 4.1.0)
#&gt;  curl                                4.3.2    2021-06-23 [1] CRAN (R 4.1.0)
#&gt;  DBI                                 1.1.1    2021-01-15 [1] CRAN (R 4.1.0)
#&gt;  dbplyr                              2.1.1    2021-04-06 [2] CRAN (R 4.1.0)
#&gt;  DelayedArray                        0.18.0   2021-05-19 [1] Bioconductor  
#&gt;  desc                                1.3.0    2021-03-05 [1] CRAN (R 4.1.0)
#&gt;  digest                              0.6.27   2020-10-24 [2] CRAN (R 4.1.0)
#&gt;  DNAcopy                             1.66.0   2021-05-19 [1] Bioconductor  
#&gt;  dplyr                               1.0.7    2021-06-18 [1] CRAN (R 4.1.0)
#&gt;  ellipsis                            0.3.2    2021-04-29 [2] CRAN (R 4.1.0)
#&gt;  evaluate                            0.14     2019-05-28 [1] CRAN (R 4.1.0)
#&gt;  fansi                               0.5.0    2021-05-25 [1] CRAN (R 4.1.0)
#&gt;  fastmap                             1.1.0    2021-01-25 [2] CRAN (R 4.1.0)
#&gt;  filelock                            1.0.2    2018-10-05 [1] CRAN (R 4.1.0)
#&gt;  fs                                  1.5.0    2020-07-31 [2] CRAN (R 4.1.0)
#&gt;  generics                            0.1.0    2020-10-31 [1] CRAN (R 4.1.0)
#&gt;  GenomeInfoDb                      * 1.28.1   2021-07-01 [1] Bioconductor  
#&gt;  GenomeInfoDbData                    1.2.6    2021-07-22 [1] Bioconductor  
#&gt;  GenomicAlignments                   1.28.0   2021-05-19 [1] Bioconductor  
#&gt;  GenomicFeatures                     1.44.0   2021-05-19 [2] Bioconductor  
#&gt;  GenomicRanges                     * 1.44.0   2021-05-19 [1] Bioconductor  
#&gt;  ggplot2                             3.3.5    2021-06-25 [2] CRAN (R 4.1.0)
#&gt;  glue                                1.4.2    2020-08-27 [2] CRAN (R 4.1.0)
#&gt;  gtable                              0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
#&gt;  highr                               0.9      2021-04-16 [1] CRAN (R 4.1.0)
#&gt;  hms                                 1.1.0    2021-05-17 [1] CRAN (R 4.1.0)
#&gt;  htmltools                           0.5.1.1  2021-01-22 [2] CRAN (R 4.1.0)
#&gt;  httpuv                              1.6.2    2021-08-18 [1] CRAN (R 4.1.0)
#&gt;  httr                                1.4.2    2020-07-20 [1] CRAN (R 4.1.0)
#&gt;  IRanges                           * 2.26.0   2021-05-19 [1] Bioconductor  
#&gt;  jquerylib                           0.1.4    2021-04-26 [1] CRAN (R 4.1.0)
#&gt;  jsonlite                            1.7.2    2020-12-09 [2] CRAN (R 4.1.0)
#&gt;  KEGGREST                            1.32.0   2021-05-19 [1] Bioconductor  
#&gt;  knitr                               1.33     2021-04-24 [1] CRAN (R 4.1.0)
#&gt;  later                               1.3.0    2021-08-18 [1] CRAN (R 4.1.0)
#&gt;  lattice                             0.20-44  2021-05-02 [2] CRAN (R 4.1.0)
#&gt;  lifecycle                           1.0.0    2021-02-15 [2] CRAN (R 4.1.0)
#&gt;  limma                               3.48.3   2021-08-10 [1] Bioconductor  
#&gt;  magrittr                            2.0.1    2020-11-17 [2] CRAN (R 4.1.0)
#&gt;  MASS                                7.3-54   2021-05-03 [2] CRAN (R 4.1.0)
#&gt;  Matrix                              1.3-3    2021-05-04 [2] CRAN (R 4.1.0)
#&gt;  MatrixGenerics                      1.4.2    2021-08-08 [1] Bioconductor  
#&gt;  matrixStats                         0.60.0   2021-07-26 [1] CRAN (R 4.1.0)
#&gt;  mclust                              5.4.7    2020-11-20 [1] CRAN (R 4.1.0)
#&gt;  memoise                             2.0.0    2021-01-26 [1] CRAN (R 4.1.0)
#&gt;  mime                                0.11     2021-06-23 [2] CRAN (R 4.1.0)
#&gt;  multtest                            2.48.0   2021-05-19 [1] Bioconductor  
#&gt;  munsell                             0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
#&gt;  org.Hs.eg.db                        3.13.0   2021-05-28 [1] Bioconductor  
#&gt;  pillar                              1.6.1    2021-05-16 [2] CRAN (R 4.1.0)
#&gt;  pkgconfig                           2.0.3    2019-09-22 [1] CRAN (R 4.1.0)
#&gt;  pkgdown                             1.6.1    2020-09-12 [1] CRAN (R 4.1.0)
#&gt;  plyr                                1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
#&gt;  png                                 0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
#&gt;  preprocessCore                      1.54.0   2021-05-19 [1] Bioconductor  
#&gt;  prettyunits                         1.1.1    2020-01-24 [1] CRAN (R 4.1.0)
#&gt;  progress                            1.2.2    2019-05-16 [1] CRAN (R 4.1.0)
#&gt;  promises                            1.2.0.1  2021-02-11 [1] CRAN (R 4.1.0)
#&gt;  purrr                               0.3.4    2020-04-17 [1] CRAN (R 4.1.0)
#&gt;  R6                                  2.5.0    2020-10-28 [2] CRAN (R 4.1.0)
#&gt;  ragg                                1.1.3    2021-06-09 [1] CRAN (R 4.1.0)
#&gt;  rappdirs                            0.3.3    2021-01-31 [2] CRAN (R 4.1.0)
#&gt;  rCGH                              * 1.22.0   2021-05-19 [1] Bioconductor  
#&gt;  Rcpp                                1.0.7    2021-07-07 [2] CRAN (R 4.1.0)
#&gt;  RCurl                               1.98-1.4 2021-08-17 [1] CRAN (R 4.1.0)
#&gt;  restfulr                            0.0.13   2017-08-06 [1] CRAN (R 4.1.0)
#&gt;  rjson                               0.2.20   2018-06-08 [1] CRAN (R 4.1.0)
#&gt;  rlang                               0.4.11   2021-04-30 [2] CRAN (R 4.1.0)
#&gt;  rmarkdown                           2.9      2021-06-15 [2] CRAN (R 4.1.0)
#&gt;  rprojroot                           2.0.2    2020-11-15 [1] CRAN (R 4.1.0)
#&gt;  Rsamtools                           2.8.0    2021-05-19 [1] Bioconductor  
#&gt;  RSQLite                             2.2.8    2021-08-21 [1] CRAN (R 4.1.0)
#&gt;  rstudioapi                          0.13     2020-11-12 [1] CRAN (R 4.1.0)
#&gt;  rtracklayer                         1.52.1   2021-08-15 [1] Bioconductor  
#&gt;  S4Vectors                         * 0.30.0   2021-05-19 [1] Bioconductor  
#&gt;  sass                                0.4.0    2021-05-12 [1] CRAN (R 4.1.0)
#&gt;  scales                              1.1.1    2020-05-11 [1] CRAN (R 4.1.0)
#&gt;  sessioninfo                       * 1.1.1    2018-11-05 [1] CRAN (R 4.1.0)
#&gt;  shiny                               1.6.0    2021-01-25 [1] CRAN (R 4.1.0)
#&gt;  stringi                             1.7.3    2021-07-16 [1] CRAN (R 4.1.0)
#&gt;  stringr                             1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
#&gt;  SummarizedExperiment                1.22.0   2021-05-19 [1] Bioconductor  
#&gt;  survival                            3.2-11   2021-04-26 [2] CRAN (R 4.1.0)
#&gt;  systemfonts                         1.0.2    2021-05-11 [1] CRAN (R 4.1.0)
#&gt;  textshaping                         0.3.5    2021-06-09 [1] CRAN (R 4.1.0)
#&gt;  tibble                              3.1.2    2021-05-16 [2] CRAN (R 4.1.0)
#&gt;  tidyselect                          1.1.1    2021-04-30 [1] CRAN (R 4.1.0)
#&gt;  TxDb.Hsapiens.UCSC.hg18.knownGene   3.2.2    2021-05-30 [1] Bioconductor  
#&gt;  TxDb.Hsapiens.UCSC.hg19.knownGene   3.2.2    2021-05-30 [1] Bioconductor  
#&gt;  TxDb.Hsapiens.UCSC.hg38.knownGene   3.13.0   2021-06-03 [1] Bioconductor  
#&gt;  utf8                                1.2.2    2021-07-24 [1] CRAN (R 4.1.0)
#&gt;  vctrs                               0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
#&gt;  withr                               2.4.2    2021-04-18 [2] CRAN (R 4.1.0)
#&gt;  xfun                                0.25     2021-08-06 [1] CRAN (R 4.1.0)
#&gt;  XML                                 3.99-0.7 2021-08-17 [1] CRAN (R 4.1.0)
#&gt;  xml2                                1.3.2    2020-04-23 [1] CRAN (R 4.1.0)
#&gt;  xtable                              1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
#&gt;  XVector                             0.32.0   2021-05-19 [1] Bioconductor  
#&gt;  yaml                                2.2.1    2020-02-01 [1] CRAN (R 4.1.0)
#&gt;  zlibbioc                            1.38.0   2021-05-19 [1] Bioconductor  
#&gt; 
#&gt; [1] /Users/mdozmorov/Library/R/x86_64/4.1/library
#&gt; [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library</code></pre>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Amemiya:2019aa" class="csl-entry">
Amemiya, Haley M, Anshul Kundaje, and Alan P Boyle. 2019. <span>“The ENCODE Blacklist: Identification of Problematic Regions of the Genome.”</span> <em>Sci Rep</em> 9 (1): 9354. <a href="https://doi.org/10.1038/s41598-019-45839-z">https://doi.org/10.1038/s41598-019-45839-z</a>.
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Mikhail Dozmorov, Eric Davis, Wancen Mu, Stuart Lee, Tim Triche, Douglas Phanstiel, Michael Love.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
